<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Quick tutorial about Genomic2DTK functions.
">
<title>Genomic2DTK - Quick start • Genomic2DTK</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genomic2DTK - Quick start">
<meta property="og:description" content="Quick tutorial about Genomic2DTK functions.
">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Genomic2DTK</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/Genomic2DTK.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/NChanard/Genomic2DTK/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Genomic2DTK - Quick start</h1>
                        <h4 data-toc-skip class="author">Nicolas
Chanard</h4>
            
            <h4 data-toc-skip class="date">2022-11-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NChanard/Genomic2DTK/blob/HEAD/vignettes/Genomic2DTK.Rmd" class="external-link"><code>vignettes/Genomic2DTK.Rmd</code></a></small>
      <div class="d-none name"><code>Genomic2DTK.Rmd</code></div>
    </div>

    
    
<style>
body {
text-align: justify}
</style>
<style type="text/css">
pre {
  max-height: 300px;
  overflow-y: auto;
}

pre[class] {
  max-height: 300px;
}
</style>
<style type="text/css">
.scroll-100 {
  max-height: 100px;
  overflow-y: auto;
  background-color: inherit;
}
</style>
<div class="section level2">
<h2 id="library">Library<a class="anchor" aria-label="anchor" href="#library"></a>
</h2>
<p>Genomic2DTK is a package that allows to perform aggregation analyses.
In a first step, these analyses consist in extracting 3D conformation
data at specific genome loci. Then, in a second step, to aggregate the
extracted signal on all the loci to visualize it. In this tutorial you
will see how to quickly perform this type of analysis. However, we
invite you to have a closer look at the documentation and other articles
that present in more depth the possibilities of Genomic2DTK which allows
for example the exploration and quantification of the signal at a single
locus or the performance of differential analysis.</p>
<div class="section level3">
<h3 id="install-libraries">Install libraries<a class="anchor" aria-label="anchor" href="#install-libraries"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"NChanard/Genomic2DTK"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-libraries">Load libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nchanard.github.io/Genomic2DTK/" class="external-link">Genomic2DTK</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="test-dataset">Test dataset<a class="anchor" aria-label="anchor" href="#test-dataset"></a>
</h2>
<p>The testing dataset is obtained from <em>Drosophila melanogaster S2
cells</em>. A first part of this data is provided directly in this
package and it consist of genomic location data (ChIP-seq and
annotations). An other part of testing dataset is downloaded directly
from the <a href="https://data.4dnucleome.org" class="external-link">4DN</a> platform and it
consists of 3D structure data of the genome (HiC maps in two different
conditions).</p>
<div class="section level3">
<h3 id="genomic-location-and-annotation-data">Genomic location and annotation data<a class="anchor" aria-label="anchor" href="#genomic-location-and-annotation-data"></a>
</h3>
<p>Subset of ChIP-seq peaks of Beaf-32 protein in wild type condition in
GRanges format (bed). These kind of data could be import in R with <a href="https://bioconductor.org/packages/release/bioc/html/rtracklayer.html" class="external-link">rtracklayer</a>
package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Beaf32_Peaks.gnr"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr class="header">
<th align="right">seq</th>
<th align="center">start</th>
<th align="center">end</th>
<th align="center">strand</th>
<th align="center">name</th>
<th align="center">score</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">2L</td>
<td align="center">35594</td>
<td align="center">35725</td>
<td align="center">*</td>
<td align="center">Beaf32_2</td>
<td align="center">76</td>
</tr>
<tr class="even">
<td align="right">2L</td>
<td align="center">47296</td>
<td align="center">47470</td>
<td align="center">*</td>
<td align="center">Beaf32_3</td>
<td align="center">44</td>
</tr>
<tr class="odd">
<td align="right">2L</td>
<td align="center">65770</td>
<td align="center">65971</td>
<td align="center">*</td>
<td align="center">Beaf32_5</td>
<td align="center">520</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="genomic-3d-structure">Genomic 3D structure<a class="anchor" aria-label="anchor" href="#genomic-3d-structure"></a>
</h3>
<p>The second part of the test data is downloaded directly from the <a href="https://data.4dnucleome.org" class="external-link">4DN</a> platform. It consists of data
from HiC experiments in .hic format (juicer format).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout <span class="op">=</span> <span class="fl">3600</span><span class="op">)</span> </span>
<span><span class="va">temp.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"HIC_DATA"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">temp.dir</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Hic.url</span> <span class="op">&lt;-</span> <span class="st">"https://4dn-open-data-public.s3.amazonaws.com/fourfront-webprod/wfoutput/7386f953-8da9-47b0-acb2-931cba810544/4DNFIOTPSS3L.hic"</span></span>
<span><span class="va">HicOutput.pth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">temp.dir</span>, <span class="st">"Control_HIC.hic"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">Hic.url</span>, <span class="va">HicOutput.pth</span>, method <span class="op">=</span> <span class="st">'curl'</span>, extra <span class="op">=</span> <span class="st">'-k'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="additional-genome-informations">Additional genome informations<a class="anchor" aria-label="anchor" href="#additional-genome-informations"></a>
</h3>
<p>General genomic informations used by the functions all along the
process are a data.frame containing chromosomes names and sized and the
binSize, corresponding to the HiC matrices resolution. We complete the
data with 2L and 2R chromosome sizes and HiC map resolution.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqlengths.num</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'2L'</span><span class="op">=</span><span class="fl">23513712</span>, <span class="st">'2R'</span><span class="op">=</span><span class="fl">25286936</span><span class="op">)</span></span>
<span><span class="va">chromSize.dtf</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  seqnames   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seqlengths.num</span> <span class="op">)</span>, </span>
<span>  seqlengths <span class="op">=</span> <span class="va">seqlengths.num</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">binSize.num</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="hic-preprocess">HiC Preprocess<a class="anchor" aria-label="anchor" href="#hic-preprocess"></a>
</h2>
<p>The package supports the preprocessing of HiC data and allows you to
access the information in a simplified way thanks to the data
structure.</p>
<div class="section level3">
<h3 id="import">Import<a class="anchor" aria-label="anchor" href="#import"></a>
</h3>
<p>Genomic2DTK can import HiC data stored in the main formats: .hic,
.cool, .mcool, .h5, .hdf5.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic.cmx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ImportHiC.html">ImportHiC</a></span><span class="op">(</span></span>
<span>  file.pth      <span class="op">=</span> <span class="va">HicOutput.pth</span>,</span>
<span>  res.num       <span class="op">=</span> <span class="va">binSize.num</span>,</span>
<span>  chrom_1.chr   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2L"</span>, <span class="st">"2R"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="balancing">Balancing<a class="anchor" aria-label="anchor" href="#balancing"></a>
</h3>
<p>Genomic2TK forces the import in the R environment of the raw counts
matrix, it is thus necessary to perform the balancing step (Defaut:
Iterative Correction) which makes it possible to correct the biases
inherent to the loci by balancing the sums of each line and of each
column to make them comparable.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic.cmx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NormalizeHiC.html">NormalizeHiC</a></span><span class="op">(</span><span class="va">hic.cmx_lst</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="expected">Expected<a class="anchor" aria-label="anchor" href="#expected"></a>
</h3>
<p>Once the bias inherent to the loci has been corrected, the bias due
to the genomic distance between two loci must be corrected. To do this,
the matrix is corrected by the expected values for each genomic
distance.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic.cmx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExpectedHiC.html">ExpectedHiC</a></span><span class="op">(</span><span class="va">hic.cmx_lst</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="hic-data-format-contactmatrix-list-cmx_lst-structure">HiC data format: ContactMatrix list (cmx_lst) structure<a class="anchor" aria-label="anchor" href="#hic-data-format-contactmatrix-list-cmx_lst-structure"></a>
</h3>
<p>Each element of the list corresponds to a ContactMatrix object
(dgCMatrix object, sparse matrix format) storing contact frequencies for
one chromosomes (cis-interactions, ex: “2L_2L”) or between two
chromosomes (trans-interactions, ex: “2L_2R”).<br>
HiC data format is based on <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/InteractionSet/inst/doc/interactions.html" class="external-link">InteractionSet</a>
package, for more details on sparse matrices format, see <a href="https://cran.r-project.org/web/packages/Matrix/Matrix.pdf" class="external-link">Matrix</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">hic.cmx_lst</span>, max.level <span class="op">=</span> <span class="fl">3</span>, give.attr <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ 2L_2L:Formal class 'ContactMatrix' [package "InteractionSet"] with 5 slots</span></span>
<span><span class="co">#&gt;   .. ..@ matrix  :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. ..@ anchor1 : int [1:23514] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;   .. ..@ anchor2 : int [1:23514] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;   .. ..@ regions :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots</span></span>
<span><span class="co">#&gt;   .. ..@ metadata:List of 10</span></span>
<span><span class="co">#&gt;  $ 2R_2R:Formal class 'ContactMatrix' [package "InteractionSet"] with 5 slots</span></span>
<span><span class="co">#&gt;   .. ..@ matrix  :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. ..@ anchor1 : int [1:25287] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;   .. ..@ anchor2 : int [1:25287] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;   .. ..@ regions :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots</span></span>
<span><span class="co">#&gt;   .. ..@ metadata:List of 10</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<p>The list object have some generals attributes. These attributes are
accessed via:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">hic.cmx_lst</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 6</span></span>
<span><span class="co">#&gt;  $ names       : chr [1:2] "2L_2L" "2R_2R"</span></span>
<span><span class="co">#&gt;  $ resolution  : num 1000</span></span>
<span><span class="co">#&gt;  $ chromSize   : tibble [2 × 3] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ name     : chr [1:2] "2L" "2R"</span></span>
<span><span class="co">#&gt;   ..$ length   : num [1:2] 23513712 25286936</span></span>
<span><span class="co">#&gt;   ..$ dimension: num [1:2] 23514 25287</span></span>
<span><span class="co">#&gt;  $ matricesKind: tibble [2 × 4] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ name     : chr [1:2] "2L_2L" "2R_2R"</span></span>
<span><span class="co">#&gt;   ..$ type     : chr [1:2] "cis" "cis"</span></span>
<span><span class="co">#&gt;   ..$ kind     : chr [1:2] "U" "U"</span></span>
<span><span class="co">#&gt;   ..$ symmetric: logi [1:2] TRUE TRUE</span></span>
<span><span class="co">#&gt;  $ mtx         : chr "o/e"</span></span>
<span><span class="co">#&gt;  $ expected    : tibble [24,362 × 2] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ distance: num [1:24362] 1 1001 2001 3001 4001 ...</span></span>
<span><span class="co">#&gt;   ..$ expected: num [1:24362] 52.3 68.7 42.8 35.5 30.4 ...</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>
<strong>names</strong> : the names of list elements
(ContactMatrix).</li>
<li>
<strong>resolution</strong> : the resolution of the HiC map.<br>
</li>
<li>
<strong>chromSize</strong> : the size of the chromosomes in the
tibble format.
<ul>
<li>
<em>seqnames</em> : the sequence name (chromosome name).<br>
</li>
<li>
<em>seqlengths</em> : the sequence length in base pairs.<br>
</li>
<li>
<em>dimension</em> : the sequence length in base bins<br>
</li>
</ul>
</li>
<li>
<strong>matricesKind</strong> : the kind of matrix that composes the
list in the tibble format.
<ul>
<li>
<em>name</em> : the matrix name.</li>
<li>
<em>type</em> : interaction type. <em>“Cis”</em> for interactions on
the same chromosome and <em>“Trans”</em> for interactions on different
chromosomes.<br>
</li>
<li>
<em>kind</em> : the matrix kind. U for upper triangle matrices, L
for lower triangle matrices, NA for rectangular or square matrices.</li>
<li>
<em>symmetric</em> : a boolean that indicates whether the matrix is
symmetric (lower triangle identical to upper triangle).<br>
</li>
</ul>
</li>
<li>
<strong>mtx</strong> : the kind of values in matrix. For exemple
observed counts, normalized counts, observed/expected, etc.</li>
<li>
<strong>expected</strong> : a tiblle of expected count as a function
of genomic distance</li>
</ol>
<p>Each contactmatrix in the list have metadata. These are accessed
via:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">hic.cmx_lst</span><span class="op">[[</span><span class="st">"2L_2L"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 10</span></span>
<span><span class="co">#&gt;  $ name         : chr "2L_2L"</span></span>
<span><span class="co">#&gt;  $ type         : chr "cis"</span></span>
<span><span class="co">#&gt;  $ kind         : chr "U"</span></span>
<span><span class="co">#&gt;  $ symmetric    : logi TRUE</span></span>
<span><span class="co">#&gt;  $ resolution   : num 1000</span></span>
<span><span class="co">#&gt;  $ removedCounts:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. ..@ i       : int [1:67995] 5 7 8 9 11 11 11 11 17 28 ...</span></span>
<span><span class="co">#&gt;   .. ..@ p       : int [1:23515] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;   .. ..@ Dim     : int [1:2] 23514 23514</span></span>
<span><span class="co">#&gt;   .. ..@ Dimnames:List of 2</span></span>
<span><span class="co">#&gt;   .. .. ..$ : NULL</span></span>
<span><span class="co">#&gt;   .. .. ..$ : NULL</span></span>
<span><span class="co">#&gt;   .. ..@ x       : num [1:67995] 1 2 3 1 2 3 3 1 1 1 ...</span></span>
<span><span class="co">#&gt;   .. ..@ factors : list()</span></span>
<span><span class="co">#&gt;  $ observed     : num [1:6540004] 41 126 34 60 90 32 77 89 74 87 ...</span></span>
<span><span class="co">#&gt;  $ normalizer   : num [1:6540004] 1.085 0.971 0.869 1.345 1.203 ...</span></span>
<span><span class="co">#&gt;  $ mtx          : chr "norm"</span></span>
<span><span class="co">#&gt;  $ expected     : num [1:6540004] 52 65.8 52 39.8 65.8 ...</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>
<strong>name</strong> : the name of the ContactMatrix.</li>
<li>
<strong>type</strong> : interaction type. <em>“Cis”</em> for
interactions on the same chromosome and <em>“Trans”</em> for
interactions on different chromosomes.</li>
<li>
<strong>kind</strong> : the matrix kind. U for upper triangle
matrices, L for lower triangle matrices, NA for rectangular or
square</li>
<li>
<strong>symmetric</strong> : a boolean that indicates whether the
matrix is symmetric (lower triangle identical to upper triangle).<br>
</li>
<li>
<strong>resolution</strong> : the resolution of the HiC map.<br>
</li>
<li>
<strong>removedCounts</strong> : A sparse matrix (dgCMatrix) of all
counts that are removed beacause there are on very low count row or very
low count columns.</li>
<li>
<strong>observed</strong> : the raw observed counts of the sparse
matrix.</li>
<li>
<strong>normalizer</strong> : the balancer vector to apply on the
raw observed counts to obtaine the normalized counts. (observed *
normalizer = normalized counts).<br>
</li>
<li>
<strong>mtx</strong> : the kind of values in matrix. For exemple
observed counts, normalized counts, observed/expected, etc.<br>
</li>
<li>
<strong>expected</strong> : the expected vector to apply on the
normalized counts to obtaine the observed/expected counts. (normalized
counts/expected = observed/expected).</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="genomic-locations-preprocess">Genomic locations Preprocess<a class="anchor" aria-label="anchor" href="#genomic-locations-preprocess"></a>
</h2>
<p>This part of the data corresponds to the positioning data on the
genome. To integrate them with the HiC data, which are represented by 2D
matrices, these data have to be processed in some way.</p>
<div class="section level3">
<h3 id="features-indexing">Features Indexing<a class="anchor" aria-label="anchor" href="#features-indexing"></a>
</h3>
<p>The first step is the indexing of the features on the genome. This
step allows the features to be splitted and grouped into bins that
correspond to the resolution of the HiC matrix.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Beaf_Index.gnr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IndexFeatures.html">IndexFeatures</a></span><span class="op">(</span></span>
<span>  gRange.gnr_lst        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Beaf <span class="op">=</span> <span class="va">Beaf32_Peaks.gnr</span><span class="op">)</span>,</span>
<span>  chromSize.dtf         <span class="op">=</span> <span class="va">chromSize.dtf</span>,</span>
<span>  binSize.num           <span class="op">=</span> <span class="va">binSize.num</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="beaf32---beaf32-putatives-pairs">Beaf32 &lt;-&gt; Beaf32 putatives pairs<a class="anchor" aria-label="anchor" href="#beaf32---beaf32-putatives-pairs"></a>
</h3>
<p>The second step is to search for pairs of bins. We put here two
distance constraints in order to limit the number of pairs kept. Thus,
only the pairs for which the peaks of Beaf32 are at least 25KB and at
most 100Kb apart are retained.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Beaf_Pairs.gni</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SearchPairs.html">SearchPairs</a></span><span class="op">(</span></span>
<span>  indexAnchor.gnr <span class="op">=</span> <span class="va">Beaf_Index.gnr</span>,</span>
<span>  minDist.num     <span class="op">=</span> <span class="st">"25KB"</span>,</span>
<span>  maxDist.num     <span class="op">=</span> <span class="st">"100KB"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="aggregation-analysis">Aggregation Analysis<a class="anchor" aria-label="anchor" href="#aggregation-analysis"></a>
</h2>
<p>Once the data has been pre-processed, the analysis can be performed
to integrate it. ## Extraction The first step is to extract the HiC data
centred on the previously formed loci pairs. The result of the
extraction is a list of matrices where each matrix corresponds to a
pair.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Beaf.mtx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn  <span class="op">=</span> <span class="va">Beaf_Pairs.gni</span>,</span>
<span>  hic.cmx_lst <span class="op">=</span> <span class="va">hic.cmx_lst</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="aggregation">Aggregation<a class="anchor" aria-label="anchor" href="#aggregation"></a>
</h3>
<p>Then we aggregate all extracted matrices in one unique matrice by
performing as default a mean.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aggreg.mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Aggregation.html">Aggregation</a></span><span class="op">(</span><span class="va">Beaf.mtx_lst</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visulatisation">Visulatisation<a class="anchor" aria-label="anchor" href="#visulatisation"></a>
</h3>
<p>Finaly, we can plot the aggregated matrix. Here we can see a red
square that correspond to the topological associated domains formed
between beaf32 loci.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>  apa.mtx <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>  title.chr <span class="op">=</span> <span class="st">"APA Beaf &lt;-&gt; Beaf"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-19-1.png" width="672"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicolas Chanard .</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
