<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Quick tutorial about Genomic2DTK functions.
">
<title>Genomic2DTK - Quick start â€¢ Genomic2DTK</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genomic2DTK - Quick start">
<meta property="og:description" content="Quick tutorial about Genomic2DTK functions.
">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Genomic2DTK</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/Genomic2DTK.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/InDepth.html">Genomic2DTK - In depth tutorial</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/NChanard/Genomic2DTK/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Genomic2DTK - Quick start</h1>
                        <h4 data-toc-skip class="author">Nicolas
Chanard</h4>
            
            <h4 data-toc-skip class="date">2022-12-07</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NChanard/Genomic2DTK/blob/HEAD/vignettes/Genomic2DTK.Rmd" class="external-link"><code>vignettes/Genomic2DTK.Rmd</code></a></small>
      <div class="d-none name"><code>Genomic2DTK.Rmd</code></div>
    </div>

    
    
<style>
body {
text-align: justify}
</style>
<style type="text/css">
pre {
    max-height: 300px;
    overflow-y: auto;
}

pre[class] {
    max-height: 300px;
}
</style>
<style type="text/css">
.scroll-100 {
    max-height: 100px;
    overflow-y: auto;
    background-color: inherit;
}
</style>
<p>Quick start from the import of HiC data to the aggregation of HiC
contact in 3D. It includes 4 steps:</p>
<ol style="list-style-type: decimal">
<li>Import HiC</li>
<li>Import genomic coordinates</li>
<li>Submatrices extractions</li>
<li>Plot and visualization</li>
</ol>
<div class="section level2">
<h2 id="requirements">Requirements<a class="anchor" aria-label="anchor" href="#requirements"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"NChanard/Genomic2DTK"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-library">Load library<a class="anchor" aria-label="anchor" href="#load-library"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nchanard.github.io/Genomic2DTK/" class="external-link">Genomic2DTK</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="test-dataset">Test dataset<a class="anchor" aria-label="anchor" href="#test-dataset"></a>
</h2>
<div class="section level3">
<h3 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h3>
<p>Data were obtained from <em>Drosophila melanogaster S2 cells</em>.
HiC test dataset: Directly downloaded from the <a href="https://data.4dnucleome.org" class="external-link">4DN</a> platform. Genomic
coordinates: ChIPseq peaks of Beaf-32 protein in wild type cells (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1278639" class="external-link">GSM1278639</a>).</p>
</div>
<div class="section level3">
<h3 id="genomic-3d-structure">Genomic 3D structure<a class="anchor" aria-label="anchor" href="#genomic-3d-structure"></a>
</h3>
<p>For a test, please download HiC data in .hic format (Juicer).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout <span class="op">=</span> <span class="fl">3600</span><span class="op">)</span> </span>
<span><span class="va">temp.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"HIC_DATA"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">temp.dir</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Hic.url</span> <span class="op">&lt;-</span> <span class="st">"https://4dn-open-data-public.s3.amazonaws.com/fourfront-webprod/wfoutput/7386f953-8da9-47b0-acb2-931cba810544/4DNFIOTPSS3L.hic"</span></span>
<span><span class="va">HicOutput.pth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">temp.dir</span>, <span class="st">"Control_HIC.hic"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">Hic.url</span>, <span class="va">HicOutput.pth</span>, method <span class="op">=</span> <span class="st">'curl'</span>, extra <span class="op">=</span> <span class="st">'-k'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="genomic-location-and-annotation-data">Genomic location and annotation data<a class="anchor" aria-label="anchor" href="#genomic-location-and-annotation-data"></a>
</h3>
<p>These kind of data can be imported in R with <a href="https://bioconductor.org/packages/release/bioc/html/rtracklayer.html" class="external-link">rtracklayer</a>
package.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Beaf32_Peaks.gnr"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr class="header">
<th align="right">seq</th>
<th align="center">start</th>
<th align="center">end</th>
<th align="center">strand</th>
<th align="center">name</th>
<th align="center">score</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">2L</td>
<td align="center">35594</td>
<td align="center">35725</td>
<td align="center">*</td>
<td align="center">Beaf32_2</td>
<td align="center">76</td>
</tr>
<tr class="even">
<td align="right">2L</td>
<td align="center">47296</td>
<td align="center">47470</td>
<td align="center">*</td>
<td align="center">Beaf32_3</td>
<td align="center">44</td>
</tr>
<tr class="odd">
<td align="right">2L</td>
<td align="center">65770</td>
<td align="center">65971</td>
<td align="center">*</td>
<td align="center">Beaf32_5</td>
<td align="center">520</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="additional-genome-informations">Additional genome informations<a class="anchor" aria-label="anchor" href="#additional-genome-informations"></a>
</h3>
<p>Required genomic informations used by the functions during the entire
pipeline are a data.frame containing chromosomes names and sized and the
binSize, corresponding to the HiC matrices at the same resolution.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqlengths.num</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'2L'</span><span class="op">=</span><span class="fl">23513712</span>, <span class="st">'2R'</span><span class="op">=</span><span class="fl">25286936</span><span class="op">)</span></span>
<span><span class="va">chromSize.dtf</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    seqnames   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seqlengths.num</span> <span class="op">)</span>, </span>
<span>    seqlengths <span class="op">=</span> <span class="va">seqlengths.num</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">binSize.num</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="import-hic">1 Import HiC<a class="anchor" aria-label="anchor" href="#import-hic"></a>
</h2>
<p>The package supports the import and normalization of HiC data.</p>
<div class="section level3">
<h3 id="import">Import<a class="anchor" aria-label="anchor" href="#import"></a>
</h3>
<p>Genomic2DTK can import HiC data stored in the main formats: .hic,
.cool, .mcool, .h5, .hdf5. The pacakage imports only the raw counts in
R. Therefore it is necessary to perform the balancing and
observed/expected correction steps.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic.cmx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ImportHiC.html">ImportHiC</a></span><span class="op">(</span></span>
<span>    file.pth      <span class="op">=</span> <span class="va">HicOutput.pth</span>,</span>
<span>    res.num       <span class="op">=</span> <span class="va">binSize.num</span>,</span>
<span>    chrom_1.chr   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2L"</span>, <span class="st">"2R"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="balancing">Balancing<a class="anchor" aria-label="anchor" href="#balancing"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic.cmx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BalanceHiC.html">BalanceHiC</a></span><span class="op">(</span><span class="va">hic.cmx_lst</span><span class="op">)</span></span>
<span><span class="co">#&gt; No trans matrix, Normalisation won't be applied on trans matrices</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="observedexpected-correction">Observed/Expected Correction<a class="anchor" aria-label="anchor" href="#observedexpected-correction"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic.cmx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OverExpectedHiC.html">OverExpectedHiC</a></span><span class="op">(</span><span class="va">hic.cmx_lst</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="import-genomic-coordinates">2 Import genomic coordinates<a class="anchor" aria-label="anchor" href="#import-genomic-coordinates"></a>
</h2>
<p>Genomic coordinates data (ChIP seq peaks or any other feature) need
to be indexed using the same referenced genome as for HiC data. Then the
genomic coordinates are paired in GInteraction objects.</p>
<div class="section level3">
<h3 id="features-indexing">Features Indexing<a class="anchor" aria-label="anchor" href="#features-indexing"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Beaf_Index.gnr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IndexFeatures.html">IndexFeatures</a></span><span class="op">(</span></span>
<span>    gRange.gnr_lst        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Beaf <span class="op">=</span> <span class="va">Beaf32_Peaks.gnr</span><span class="op">)</span>,</span>
<span>    chromSize.dtf         <span class="op">=</span> <span class="va">chromSize.dtf</span>,</span>
<span>    binSize.num           <span class="op">=</span> <span class="va">binSize.num</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="beaf32---beaf32-putatives">Beaf32 &lt;-&gt; Beaf32 putatives<a class="anchor" aria-label="anchor" href="#beaf32---beaf32-putatives"></a>
</h3>
<p>Constraints for the distance between interaction sites are defined
here in order to limit the number of pairs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Beaf_Pairs.gni</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SearchPairs.html">SearchPairs</a></span><span class="op">(</span></span>
<span>    indexAnchor.gnr <span class="op">=</span> <span class="va">Beaf_Index.gnr</span>,</span>
<span>    minDist.num     <span class="op">=</span> <span class="st">"25KB"</span>,</span>
<span>    maxDist.num     <span class="op">=</span> <span class="st">"100KB"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="submatrices-extractions">3 Submatrices extractions<a class="anchor" aria-label="anchor" href="#submatrices-extractions"></a>
</h2>
<p>Once data have been imported, interactions are extracted out of the
pairs of genomic coordinates.</p>
<p><img src="images/Extractions_of_LRI.png"><!-- --></p>
<div class="section level3">
<h3 id="extraction">Extraction<a class="anchor" aria-label="anchor" href="#extraction"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Beaf.mtx_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>    feature.gn  <span class="op">=</span> <span class="va">Beaf_Pairs.gni</span>,</span>
<span>    hic.cmx_lst <span class="op">=</span> <span class="va">hic.cmx_lst</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="plot-and-visualization">4 Plot and visualization<a class="anchor" aria-label="anchor" href="#plot-and-visualization"></a>
</h2>
<p>Submatrices are aggregated as sum, average or median. Then,
aggregated matrix is plotted as a heatmap of contact frequencies (in the
example, contacts surounding Beaf-32 sites).</p>
<div class="section level3">
<h3 id="aggregation">Aggregation<a class="anchor" aria-label="anchor" href="#aggregation"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aggreg.mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Aggregation.html">Aggregation</a></span><span class="op">(</span><span class="va">Beaf.mtx_lst</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visulatisation">Visulatisation<a class="anchor" aria-label="anchor" href="#visulatisation"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr <span class="op">=</span> <span class="st">"APA Beaf &lt;-&gt; Beaf"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-17-1.png" width="672"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] Genomic2DTK_0.99.0          InteractionSet_1.24.0      </span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.26.1 Biobase_2.56.0             </span></span>
<span><span class="co">#&gt;  [5] MatrixGenerics_1.8.1        matrixStats_0.63.0         </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.48.0        GenomeInfoDb_1.32.4        </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.30.1              S4Vectors_0.34.0           </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.42.0        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] Rcpp_1.0.9             lattice_0.20-45        tidyr_1.2.1           </span></span>
<span><span class="co">#&gt;  [4] rprojroot_2.0.3        digest_0.6.30          utf8_1.2.2            </span></span>
<span><span class="co">#&gt;  [7] R6_2.5.1               evaluate_0.18          ggplot2_3.4.0         </span></span>
<span><span class="co">#&gt; [10] highr_0.9              pillar_1.8.1           zlibbioc_1.42.0       </span></span>
<span><span class="co">#&gt; [13] rlang_1.0.6            jquerylib_0.1.4        Matrix_1.5-1          </span></span>
<span><span class="co">#&gt; [16] rmarkdown_2.18         pkgdown_2.0.6          labeling_0.4.2        </span></span>
<span><span class="co">#&gt; [19] textshaping_0.3.6      desc_1.4.2             BiocParallel_1.30.4   </span></span>
<span><span class="co">#&gt; [22] stringr_1.5.0          munsell_0.5.0          RCurl_1.98-1.9        </span></span>
<span><span class="co">#&gt; [25] DelayedArray_0.22.0    compiler_4.2.2         xfun_0.35             </span></span>
<span><span class="co">#&gt; [28] pkgconfig_2.0.3        systemfonts_1.0.4      strawr_0.0.9          </span></span>
<span><span class="co">#&gt; [31] htmltools_0.5.3        tidyselect_1.2.0       tibble_3.1.8          </span></span>
<span><span class="co">#&gt; [34] GenomeInfoDbData_1.2.8 codetools_0.2-18       fansi_1.0.3           </span></span>
<span><span class="co">#&gt; [37] dplyr_1.0.10           withr_2.5.0            bitops_1.0-7          </span></span>
<span><span class="co">#&gt; [40] grid_4.2.2             gtable_0.3.1           jsonlite_1.8.3        </span></span>
<span><span class="co">#&gt; [43] lifecycle_1.0.3        magrittr_2.0.3         scales_1.2.1          </span></span>
<span><span class="co">#&gt; [46] cli_3.4.1              stringi_1.7.8          cachem_1.0.6          </span></span>
<span><span class="co">#&gt; [49] farver_2.1.1           XVector_0.36.0         fs_1.5.2              </span></span>
<span><span class="co">#&gt; [52] bslib_0.4.1            ellipsis_0.3.2         ragg_1.2.4            </span></span>
<span><span class="co">#&gt; [55] generics_0.1.3         vctrs_0.5.1            tools_4.2.2           </span></span>
<span><span class="co">#&gt; [58] glue_1.6.2             purrr_0.3.5            parallel_4.2.2        </span></span>
<span><span class="co">#&gt; [61] fastmap_1.1.0          yaml_2.3.6             colorspace_2.0-3      </span></span>
<span><span class="co">#&gt; [64] memoise_2.0.1          knitr_1.41             sass_0.4.4</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicolas Chanard, David Depierre, Olivier Cuvier.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
