<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="In-depth tutorial about Genomic2DTK functions.
">
<title>Genomic2DTK - In-depth tutorial • Genomic2DTK</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genomic2DTK - In-depth tutorial">
<meta property="og:description" content="In-depth tutorial about Genomic2DTK functions.
">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Genomic2DTK</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/Genomic2DTK.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/NChanard/Genomic2DTK/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="Genomic2DTK_files/kePrint-0.0.1/kePrint.js"></script><link href="Genomic2DTK_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Genomic2DTK - In-depth tutorial</h1>
                        <h4 data-toc-skip class="author">Nicolas
Chanard</h4>
            
            <h4 data-toc-skip class="date">2022-11-07</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NChanard/Genomic2DTK/blob/HEAD/vignettes/Genomic2DTK.Rmd" class="external-link"><code>vignettes/Genomic2DTK.Rmd</code></a></small>
      <div class="d-none name"><code>Genomic2DTK.Rmd</code></div>
    </div>

    
    
<style>
body {
text-align: justify}
</style>
<style type="text/css">
pre {
  max-height: 300px;
  overflow-y: auto;
}

pre[class] {
  max-height: 300px;
}
</style>
<style type="text/css">
.scroll-100 {
  max-height: 100px;
  overflow-y: auto;
  background-color: inherit;
}
</style>
<div class="section level2">
<h2 id="library">Library<a class="anchor" aria-label="anchor" href="#library"></a>
</h2>
<div class="section level3">
<h3 id="install-libraries">Install libraries<a class="anchor" aria-label="anchor" href="#install-libraries"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"NChanard/Genomic2DTK"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"NChanard/GenomicED"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-libraries">Load libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nchanard.github.io/Genomic2DTK/" class="external-link">Genomic2DTK</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: InteractionSet</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span></span>
<span><span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span></span>
<span><span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span></span>
<span><span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nchanard.github.io/GenomicED/" class="external-link">GenomicED</a></span><span class="op">)</span></span></code></pre></div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="test-dataset">Test dataset<a class="anchor" aria-label="anchor" href="#test-dataset"></a>
</h2>
<p>Test dataset originates from <em>Drosophila melanogaster</em>. It
contains HiC data of chromosomes 2L and 2R in control and BEAF32 KD
conditions, a subset of Beaf32 ChIP-seq peaks and a subset of TSS from
<a href="https://bioconductor.org/packages/release/data/annotation/html/TxDb.Dmelanogaster.UCSC.dm6.ensGene.html" class="external-link">TxDb.Dmelanogaster.UCSC.dm6.ensGene</a>.</p>
<div class="section level3">
<h3 id="hic-test-dataset">HiC test dataset<a class="anchor" aria-label="anchor" href="#hic-test-dataset"></a>
</h3>
<div class="section level4">
<h4 id="hic_ctrl-gnp">HiC_ctrl.gnp<a class="anchor" aria-label="anchor" href="#hic_ctrl-gnp"></a>
</h4>
<p>HiC data in control condition (depletion of inexistent Luciferase
(Luc) protein) in the GRangePairs format (bedpe) with 1KBp
resolution.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HiC_ctrl.gnp"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="4">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
First
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="4">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Second
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Metadata
</div>
</th>
</tr>
<tr>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
score
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
74001
</td>
<td style="text-align:center;">
75000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
73000_74000
</td>
<td style="text-align:center;">
0.9
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
74001
</td>
<td style="text-align:center;">
75000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
74001
</td>
<td style="text-align:center;">
75000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
74000_74000
</td>
<td style="text-align:center;">
1.8
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
75001
</td>
<td style="text-align:center;">
76000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
75001
</td>
<td style="text-align:center;">
76000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
75000_75000
</td>
<td style="text-align:center;">
1.7
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level4">
<h4 id="hic-gnp">HiC.gnp<a class="anchor" aria-label="anchor" href="#hic-gnp"></a>
</h4>
<p>HiC data in Beaf-32 depleted condition in the GRangePairs format
(bedpe) with 1KBp resolution.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HiC.gnp"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="4">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
First
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="4">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Second
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Metadata
</div>
</th>
</tr>
<tr>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
score
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
73000_73000
</td>
<td style="text-align:center;">
2.3
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
74001
</td>
<td style="text-align:center;">
75000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
73000_74000
</td>
<td style="text-align:center;">
1.1
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
75001
</td>
<td style="text-align:center;">
76000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
73000_75000
</td>
<td style="text-align:center;">
2.9
</td>
</tr>
</tbody>
</table></details>
</div>
</div>
<div class="section level3">
<h3 id="anchors-baits-and-constraints-1d-and-2d-coordinates-test-dataset">Anchors, baits and constraints: 1D and 2D coordinates test
dataset<a class="anchor" aria-label="anchor" href="#anchors-baits-and-constraints-1d-and-2d-coordinates-test-dataset"></a>
</h3>
<div class="section level4">
<h4 id="chip-seq-peaks-anchors_peaks-gnr">ChIP-seq peaks: anchors_Peaks.gnr<a class="anchor" aria-label="anchor" href="#chip-seq-peaks-anchors_peaks-gnr"></a>
</h4>
<p>ChIPseq data of Beaf-32 protein in wild type condition in GRanges
format (bed).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"anchors_Peaks.gnr"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
score
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
35594
</td>
<td style="text-align:center;">
35725
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
Beaf32_2
</td>
<td style="text-align:center;">
76
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
47296
</td>
<td style="text-align:center;">
47470
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
Beaf32_3
</td>
<td style="text-align:center;">
44
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
65770
</td>
<td style="text-align:center;">
65971
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
Beaf32_5
</td>
<td style="text-align:center;">
520
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level4">
<h4 id="reference-annontation-baits_peaks-gnr">Reference annontation: baits_Peaks.gnr<a class="anchor" aria-label="anchor" href="#reference-annontation-baits_peaks-gnr"></a>
</h4>
<p>Data from the UCSC data base.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"baits_Peaks.gnr"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
class
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
71757
</td>
<td style="text-align:center;">
71757
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
FBgn0031213
</td>
<td style="text-align:center;">
active
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
76348
</td>
<td style="text-align:center;">
76348
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
FBgn0031214
</td>
<td style="text-align:center;">
inactive
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
106903
</td>
<td style="text-align:center;">
106903
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
FBgn0005278
</td>
<td style="text-align:center;">
active
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level4">
<h4 id="domains-annotation-domains-gnr">Domains annotation: domains.gnr<a class="anchor" aria-label="anchor" href="#domains-annotation-domains-gnr"></a>
</h4>
<p>Data from TAD calling on HIC data in wild type (bed or bedpe format)
(<a href="https://doi.org/10.1038/s41467-017-02525-w" class="external-link">F. Ramirez,
2018</a>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"domains.gnr"</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
score
</th>
<th style="text-align:center;">
class
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73104
</td>
<td style="text-align:center;">
94543
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
Tad_1
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
active
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
94544
</td>
<td style="text-align:center;">
102930
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
Tad_2
</td>
<td style="text-align:center;">
8
</td>
<td style="text-align:center;">
active
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
102931
</td>
<td style="text-align:center;">
121473
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
Tad_3
</td>
<td style="text-align:center;">
8
</td>
<td style="text-align:center;">
active
</td>
</tr>
</tbody>
</table></details>
</div>
</div>
<div class="section level3">
<h3 id="additional-genome-informations">Additional genome informations<a class="anchor" aria-label="anchor" href="#additional-genome-informations"></a>
</h3>
<p>General genomic informations used by the functions all along the
process are a data.frame containing chromosomes names and sized and the
binSize, corresponding to the HiC matrices resolution. We complete the
data with 2L and 2R chromosome sizes and HiC map resolution.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seqlengths.num</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'2L'</span><span class="op">=</span><span class="fl">23513712</span>, <span class="st">'2R'</span><span class="op">=</span><span class="fl">25286936</span><span class="op">)</span></span>
<span><span class="va">chromSize.dtf</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  seqnames   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seqlengths.num</span> <span class="op">)</span>, </span>
<span>  seqlengths <span class="op">=</span> <span class="va">seqlengths.num</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">binSize.num</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span></code></pre></div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="convert-hic-bedpe-this-section-will-be-replaced-by-import-hic-function">Convert HiC bedpe // this section will be replaced by Import.hic
function<a class="anchor" aria-label="anchor" href="#convert-hic-bedpe-this-section-will-be-replaced-by-import-hic-function"></a>
</h2>
<div class="section level3">
<h3 id="hic-control">HiC Control<a class="anchor" aria-label="anchor" href="#hic-control"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">HiC_ctrl.cmx_lst</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="hic-under-beaf-depletion">HiC under Beaf depletion<a class="anchor" aria-label="anchor" href="#hic-under-beaf-depletion"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">HiC.cmx_lst</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="hic-data-format-contact-matrix-list-cmx_lst-structure">HiC data format: Contact matrix list (cmx_lst) structure<a class="anchor" aria-label="anchor" href="#hic-data-format-contact-matrix-list-cmx_lst-structure"></a>
</h3>
<p>Each element of the list corresponds to a Contactmatrix object
(dgCMatrix object, sparse matrix format) storing contact frequencies for
one chromosomes (cis-interactions, ex: “2L_2L”) or between two
chromosomes (trans-interactions, ex: “2L_2R”).<br>
HiC data format is based on <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/InteractionSet/inst/doc/interactions.html" class="external-link">InteractionSet</a>
package, for more details on sparse matrices format, see <a href="https://cran.r-project.org/web/packages/Matrix/Matrix.pdf" class="external-link">Matrix</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HiC_ctrl.cmx_lst</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="co">#&gt; $`2L_2L`</span></span>
<span><span class="co">#&gt; class: ContactMatrix </span></span>
<span><span class="co">#&gt; dim: 23514 23514 </span></span>
<span><span class="co">#&gt; type: dgCMatrix </span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; metadata(5): name type kind symmetric resolution</span></span>
<span><span class="co">#&gt; regions: 23514</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2R_2R`</span></span>
<span><span class="co">#&gt; class: ContactMatrix </span></span>
<span><span class="co">#&gt; dim: 25287 25287 </span></span>
<span><span class="co">#&gt; type: dgCMatrix </span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; metadata(5): name type kind symmetric resolution</span></span>
<span><span class="co">#&gt; regions: 25287</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">HiC_ctrl.cmx_lst</span><span class="op">[[</span><span class="st">"2L_2L"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "dgCMatrix"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "Matrix"</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">HiC_ctrl.cmx_lst</span><span class="op">[[</span><span class="st">"2L_2L"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span>
<span><span class="co">#&gt;   ..@ i       : int [1:617185] 73 74 75 75 74 77 78 78 79 73 ...</span></span>
<span><span class="co">#&gt;   ..@ p       : int [1:23515] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;   ..@ Dim     : int [1:2] 23514 23514</span></span>
<span><span class="co">#&gt;   ..@ Dimnames:List of 2</span></span>
<span><span class="co">#&gt;   .. ..$ : NULL</span></span>
<span><span class="co">#&gt;   .. ..$ : NULL</span></span>
<span><span class="co">#&gt;   ..@ x       : num [1:617185] 0.899 1.772 1.729 2.185 2.752 ...</span></span>
<span><span class="co">#&gt;   ..@ factors : list()</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HiC_ctrl.cmx_lst</span><span class="op">[[</span><span class="st">"2L_2L"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">matrix</span><span class="op">[</span><span class="fl">101</span><span class="op">:</span><span class="fl">110</span>,<span class="fl">101</span><span class="op">:</span><span class="fl">110</span><span class="op">]</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:center;">
bin 101
</th>
<th style="text-align:center;">
bin 102
</th>
<th style="text-align:center;">
bin 103
</th>
<th style="text-align:center;">
bin 104
</th>
<th style="text-align:center;">
bin 105
</th>
<th style="text-align:center;">
bin 106
</th>
<th style="text-align:center;">
bin 107
</th>
<th style="text-align:center;">
bin 108
</th>
<th style="text-align:center;">
bin 109
</th>
<th style="text-align:center;">
bin 110
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
bin 101
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.9
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
2.2
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 102
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
1.6
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
1.3
</td>
<td style="text-align:center;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 103
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
2.1
</td>
<td style="text-align:center;">
2.1
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 104
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
3.6
</td>
<td style="text-align:center;">
0.9
</td>
<td style="text-align:center;">
2.1
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 105
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
2.3
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
3.3
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 106
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 107
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
2.3
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 108
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.6
</td>
<td style="text-align:center;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 109
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
1.5
</td>
<td style="text-align:center;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
bin 110
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
0.0
</td>
<td style="text-align:center;">
1.7
</td>
</tr>
</tbody>
</table></details><p>Acces to Contact matrix list attributes:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">HiC_ctrl.cmx_lst</span><span class="op">)</span></span>
<span><span class="co">#&gt; $names</span></span>
<span><span class="co">#&gt; [1] "2L_2L" "2R_2R"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $resolution</span></span>
<span><span class="co">#&gt; [1] 1000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $chromSize</span></span>
<span><span class="co">#&gt;   seqnames seqlengths dimension</span></span>
<span><span class="co">#&gt; 1       2L   23513712     23514</span></span>
<span><span class="co">#&gt; 2       2R   25286936     25287</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $matricesKind</span></span>
<span><span class="co">#&gt;    name type kind symmetric</span></span>
<span><span class="co">#&gt; 1 2L_2L  cis    U      TRUE</span></span>
<span><span class="co">#&gt; 2 2R_2R  cis    U      TRUE</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<p>Acces to Contact matrix metadata:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">HiC_ctrl.cmx_lst</span><span class="op">[[</span><span class="st">"2L_2L"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; $name</span></span>
<span><span class="co">#&gt; [1] "2L_2L"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $type</span></span>
<span><span class="co">#&gt; [1] "cis"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $kind</span></span>
<span><span class="co">#&gt; [1] "U"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $symmetric</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $resolution</span></span>
<span><span class="co">#&gt; [1] 1000</span></span>
<span><span class="co">#&gt;</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>
<strong>resolution</strong> : the resolution of the HiC map.<br>
</li>
<li>
<strong>chromSize</strong> : the size of the chromosomes in the
tibble format.
<ul>
<li>
<em>seqnames</em> : the sequence name (chromosome name).<br>
</li>
<li>
<em>seqlengths</em> : the sequence length in base pairs.<br>
</li>
<li>
<em>dimension</em> : the sequence length in base bins<br>
</li>
</ul>
</li>
<li>
<strong>matricesKind</strong> : the kind of matrix that composes the
list in the tibble format.
<ul>
<li>
<em>name</em> : the matrix name.</li>
<li>
<em>type</em> : interaction type. <em>“Cis”</em> for interactions on
the same chromosome and <em>“Trans”</em> for interactions on different
chromosomes.<br>
</li>
<li>
<em>kind</em> : the matrix kind. U for upper triangle matrices, L
for lower triangle matrices, NA for rectangular or square matrices.
See<br>
</li>
<li>
<em>symmetric</em> : a boolean that indicates whether the matrix is
symmetric (lower triangle identical to upper triangle).</li>
</ul>
</li>
</ol>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="genomic-features-indexing">Genomic Features Indexing<a class="anchor" aria-label="anchor" href="#genomic-features-indexing"></a>
</h2>
<div class="section level3">
<h3 id="indexing">Indexing<a class="anchor" aria-label="anchor" href="#indexing"></a>
</h3>
<div class="section level4">
<h4 id="indexing-example-1-anchors-from-beaf32-chip-seq-binding-sites-peak-calling-output-bed-file">Indexing example 1: Anchors from Beaf32 ChIP-seq binding sites (peak
calling output bed file)<a class="anchor" aria-label="anchor" href="#indexing-example-1-anchors-from-beaf32-chip-seq-binding-sites-peak-calling-output-bed-file"></a>
</h4>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anchors_Index.gnr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IndexFeatures.html">IndexFeatures</a></span><span class="op">(</span></span>
<span>  gRange.gnr_lst        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Beaf<span class="op">=</span><span class="va">anchors_Peaks.gnr</span><span class="op">)</span>, </span>
<span>  constraint.gnr        <span class="op">=</span> <span class="va">domains.gnr</span>,</span>
<span>  chromSize.dtf         <span class="op">=</span> <span class="va">chromSize.dtf</span>,</span>
<span>  binSize.num           <span class="op">=</span> <span class="va">binSize.num</span>,</span>
<span>  variablesName.chr_vec <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  method.chr            <span class="op">=</span> <span class="st">"max"</span>,</span>
<span>  cores.num             <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose.bln           <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:left;">
seqnames
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
width
</th>
<th style="text-align:left;">
strand
</th>
<th style="text-align:left;">
name
</th>
<th style="text-align:left;">
bin
</th>
<th style="text-align:left;">
constraint
</th>
<th style="text-align:right;">
Beaf.score
</th>
<th style="text-align:left;">
Beaf.name
</th>
<th style="text-align:right;">
Beaf.bln
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
73001
</td>
<td style="text-align:right;">
74000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:74:Tad_1
</td>
<td style="text-align:left;">
2L:74
</td>
<td style="text-align:left;">
Tad_1
</td>
<td style="text-align:right;">
205
</td>
<td style="text-align:left;">
Beaf32_8
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
74001
</td>
<td style="text-align:right;">
75000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:75:Tad_1
</td>
<td style="text-align:left;">
2L:75
</td>
<td style="text-align:left;">
Tad_1
</td>
<td style="text-align:right;">
205
</td>
<td style="text-align:left;">
Beaf32_8
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
102001
</td>
<td style="text-align:right;">
103000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:103:Tad_2
</td>
<td style="text-align:left;">
2L:103
</td>
<td style="text-align:left;">
Tad_2
</td>
<td style="text-align:right;">
1830
</td>
<td style="text-align:left;">
Beaf32_11
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level4">
<h4 id="indexing-example-2-baits-from-tss-transcription-start-sites-reference-annotation">Indexing example 2: Baits from TSS (transcription start sites
reference annotation)<a class="anchor" aria-label="anchor" href="#indexing-example-2-baits-from-tss-transcription-start-sites-reference-annotation"></a>
</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">baits_Index.gnr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IndexFeatures.html">IndexFeatures</a></span><span class="op">(</span></span>
<span>  gRange.gnr_lst        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Tss<span class="op">=</span><span class="va">baits_Peaks.gnr</span><span class="op">)</span>,</span>
<span>  constraint.gnr        <span class="op">=</span> <span class="va">domains.gnr</span>,</span>
<span>  chromSize.dtf         <span class="op">=</span> <span class="va">chromSize.dtf</span>,</span>
<span>  binSize.num           <span class="op">=</span> <span class="va">binSize.num</span>,</span>
<span>  variablesName.chr_vec <span class="op">=</span> <span class="st">"score"</span>,</span>
<span>  method.chr            <span class="op">=</span> <span class="st">"max"</span>,</span>
<span>  cores.num             <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose.bln           <span class="op">=</span> <span class="cn">FALSE</span> </span>
<span>  <span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:left;">
seqnames
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
width
</th>
<th style="text-align:left;">
strand
</th>
<th style="text-align:left;">
name
</th>
<th style="text-align:left;">
bin
</th>
<th style="text-align:left;">
constraint
</th>
<th style="text-align:left;">
Tss.class
</th>
<th style="text-align:left;">
Tss.name
</th>
<th style="text-align:right;">
Tss.bln
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
76001
</td>
<td style="text-align:right;">
77000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:77:Tad_1
</td>
<td style="text-align:left;">
2L:77
</td>
<td style="text-align:left;">
Tad_1
</td>
<td style="text-align:left;">
inactive
</td>
<td style="text-align:left;">
FBgn0031214
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
106001
</td>
<td style="text-align:right;">
107000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:107:Tad_3
</td>
<td style="text-align:left;">
2L:107
</td>
<td style="text-align:left;">
Tad_3
</td>
<td style="text-align:left;">
active
</td>
<td style="text-align:left;">
FBgn0026787, FBgn0005278
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
116001
</td>
<td style="text-align:right;">
117000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:117:Tad_3
</td>
<td style="text-align:left;">
2L:117
</td>
<td style="text-align:left;">
Tad_3
</td>
<td style="text-align:left;">
active
</td>
<td style="text-align:left;">
FBgn0031219
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table></details>
</div>
</div>
<div class="section level3">
<h3 id="filter-or-subset-indexed-features">Filter or subset indexed features:<a class="anchor" aria-label="anchor" href="#filter-or-subset-indexed-features"></a>
</h3>
<p>By using features names and bin IDs, it is possible to filter or
subset indexed features. Example: Subset indexed TSS that are not in the
same bin than an indexed Beaf32 peak.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">non_Overlaps.ndx</span> <span class="op">&lt;-</span> <span class="fu">SuperTK</span><span class="fu">::</span><span class="fu"><a href="https://nchanard.github.io/DevTK/reference/NotIn.html" class="external-link">NotIn</a></span><span class="op">(</span><span class="va">baits_Index.gnr</span><span class="op">$</span><span class="va">bin</span>,<span class="va">anchors_Index.gnr</span><span class="op">$</span><span class="va">bin</span><span class="op">)</span></span>
<span><span class="va">baits_Index.gnr</span> <span class="op">&lt;-</span> <span class="va">baits_Index.gnr</span><span class="op">[</span><span class="va">non_Overlaps.ndx</span>,<span class="op">]</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:left;">
seqnames
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
width
</th>
<th style="text-align:left;">
strand
</th>
<th style="text-align:left;">
name
</th>
<th style="text-align:left;">
bin
</th>
<th style="text-align:left;">
constraint
</th>
<th style="text-align:left;">
Tss.class
</th>
<th style="text-align:left;">
Tss.name
</th>
<th style="text-align:right;">
Tss.bln
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
76001
</td>
<td style="text-align:right;">
77000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:77:Tad_1
</td>
<td style="text-align:left;">
2L:77
</td>
<td style="text-align:left;">
Tad_1
</td>
<td style="text-align:left;">
inactive
</td>
<td style="text-align:left;">
FBgn0031214
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
106001
</td>
<td style="text-align:right;">
107000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:107:Tad_3
</td>
<td style="text-align:left;">
2L:107
</td>
<td style="text-align:left;">
Tad_3
</td>
<td style="text-align:left;">
active
</td>
<td style="text-align:left;">
FBgn0026787, FBgn0005278
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L
</td>
<td style="text-align:right;">
116001
</td>
<td style="text-align:right;">
117000
</td>
<td style="text-align:right;">
1000
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:left;">
2L:117:Tad_3
</td>
<td style="text-align:left;">
2L:117
</td>
<td style="text-align:left;">
Tad_3
</td>
<td style="text-align:left;">
active
</td>
<td style="text-align:left;">
FBgn0031219
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="tips">Tips<a class="anchor" aria-label="anchor" href="#tips"></a>
</h3>
<ol style="list-style-type: decimal">
<li>It is possible to index multiple features in the same time by
submitting a named list of GRanges. Names given in the list of GRanges
are then used to filter indexed features and pairs.<br>
</li>
<li>If contraint.gnr is set, anchors and baits will be paired when
located within the same region only. If contraint.gnr is NULL,
chromosomes are used as constraints.<br>
</li>
<li>When multiples ranges are in a same bin (ex: 3 ChIP-seq peaks in the
same 10kb bin), associated numeric variables in metadata
(<code>variablesName.chr_vec</code>) can be summarized according to the
defined method (<code>method.chr</code>), Example: Maximum peak score of
the bin is kept in metadata column <code>score</code>.</li>
</ol>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="search-pairs">Search Pairs<a class="anchor" aria-label="anchor" href="#search-pairs"></a>
</h2>
<div class="section level3">
<h3 id="pairing">Pairing<a class="anchor" aria-label="anchor" href="#pairing"></a>
</h3>
<p>SearchPairs function takes as input one or two indexed features and
returns all putative pairs within the same constraint (ex: wihtin the
same TAD). If only one indexed features is set in indexAnchor.gnr,
SearchPairs will return symetrical homotypic pairs A&lt;-&gt;A, if
indexAnchor.gnr and indexBait.gnr are set, it will return asymetrical
heterotypic pairs A&lt;-&gt;B</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">interactions.gni</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SearchPairs.html">SearchPairs</a></span><span class="op">(</span></span>
<span>    indexAnchor.gnr <span class="op">=</span> <span class="va">anchors_Index.gnr</span>,</span>
<span>    indexBait.gnr   <span class="op">=</span> <span class="va">baits_Index.gnr</span>,</span>
<span>    minDist.num     <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>    maxDist.num     <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    cores.num       <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    verbose.bln     <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="7">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Ranges
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="15">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Metadata
</div>
</th>
</tr>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Names
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
First
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Second
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="5">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Interaction
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="5">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Anchor
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="5">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Bait
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
constraint
</th>
<th style="text-align:center;">
distance
</th>
<th style="text-align:center;">
orientation
</th>
<th style="text-align:center;">
submatrix.name
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
bin
</th>
<th style="text-align:center;">
Beaf.name
</th>
<th style="text-align:center;">
Beaf.score
</th>
<th style="text-align:center;">
Beaf.bln
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
bin
</th>
<th style="text-align:center;">
Tss.name
</th>
<th style="text-align:center;">
Tss.class
</th>
<th style="text-align:center;">
Tss.bln
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
2L:74_2L:77
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
76001
</td>
<td style="text-align:center;">
77000
</td>
<td style="text-align:center;">
2L:74_2L:77
</td>
<td style="text-align:center;">
Tad_1
</td>
<td style="text-align:center;">
3000
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
2L:74_2L:77
</td>
<td style="text-align:center;">
2L:74:Tad_1
</td>
<td style="text-align:center;">
2L:74
</td>
<td style="text-align:center;">
Beaf32_8
</td>
<td style="text-align:center;">
205
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:77:Tad_1
</td>
<td style="text-align:center;">
2L:77
</td>
<td style="text-align:center;">
FBgn0031214
</td>
<td style="text-align:center;">
inactive
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:75_2L:77
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
74001
</td>
<td style="text-align:center;">
75000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
76001
</td>
<td style="text-align:center;">
77000
</td>
<td style="text-align:center;">
2L:75_2L:77
</td>
<td style="text-align:center;">
Tad_1
</td>
<td style="text-align:center;">
2000
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
2L:75_2L:77
</td>
<td style="text-align:center;">
2L:75:Tad_1
</td>
<td style="text-align:center;">
2L:75
</td>
<td style="text-align:center;">
Beaf32_8
</td>
<td style="text-align:center;">
205
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:77:Tad_1
</td>
<td style="text-align:center;">
2L:77
</td>
<td style="text-align:center;">
FBgn0031214
</td>
<td style="text-align:center;">
inactive
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:115_2L:107
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
114001
</td>
<td style="text-align:center;">
115000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
106001
</td>
<td style="text-align:center;">
107000
</td>
<td style="text-align:center;">
2L:115_2L:107
</td>
<td style="text-align:center;">
Tad_3
</td>
<td style="text-align:center;">
8000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:107_2L:115
</td>
<td style="text-align:center;">
2L:115:Tad_3
</td>
<td style="text-align:center;">
2L:115
</td>
<td style="text-align:center;">
Beaf32_14
</td>
<td style="text-align:center;">
1707
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:107:Tad_3
</td>
<td style="text-align:center;">
2L:107
</td>
<td style="text-align:center;">
FBgn0026787, FBgn0005278
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
1
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level3">
<h3 id="tips-1">Tips<a class="anchor" aria-label="anchor" href="#tips-1"></a>
</h3>
<ol style="list-style-type: decimal">
<li>If <code>indexBait.gnr</code> is NULL, SearchPairs will return
homotypic pairs with <code>indexAnchor.gnr</code>.</li>
<li>Minimum and maximum distances between pairs’ anchors can be set.
Note that it is also possible to filter pairs within a specific distance
during following steps.</li>
</ol>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="extractions">Extractions<a class="anchor" aria-label="anchor" href="#extractions"></a>
</h2>
<div class="section level3">
<h3 id="case-1-long-range-interactions-between-two-distal-anchors-">Case 1: Long-range interactions between two distal anchors.<a class="anchor" aria-label="anchor" href="#case-1-long-range-interactions-between-two-distal-anchors-"></a>
</h3>
<p><img src="images/Extractions_of_LRI.png"><!-- --></p>
<div class="section level4">
<h4 id="interactions-defined-with-ginteraction-or-pairs-granges-">Interactions defined with GInteraction or Pairs GRanges.<a class="anchor" aria-label="anchor" href="#interactions-defined-with-ginteraction-or-pairs-granges-"></a>
</h4>
<p>In extracted matrices, the middle of Y axis corresponds to the center
of the first element and interact with the center of second element in
the middle of X axis.</p>
<p><img src="images/LRI_GInteractions.png"><!-- --></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">interactions_PFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn         <span class="op">=</span> <span class="va">interactions.gni</span>,</span>
<span>  hic.cmx_lst        <span class="op">=</span> <span class="va">HiC_ctrl.cmx_lst</span>,</span>
<span>  referencePoint.chr <span class="op">=</span> <span class="st">"pf"</span>,</span>
<span>  matriceDim.num     <span class="op">=</span> <span class="fl">41</span>,</span>
<span>  cores.num          <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose.bln        <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="interactions-defined-with-granges-">Interactions defined with GRanges.<a class="anchor" aria-label="anchor" href="#interactions-defined-with-granges-"></a>
</h4>
<p>The middle of Y axis corresponds to the start of the range and
interact with the middle of X axis which corresponds with the end of the
ranges.</p>
<p><img src="images/LRI_GRanges.png"><!-- --></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains_PFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn         <span class="op">=</span> <span class="va">domains.gnr</span>,</span>
<span>  hic.cmx_lst        <span class="op">=</span> <span class="va">HiC_ctrl.cmx_lst</span>,</span>
<span>  referencePoint.chr <span class="op">=</span> <span class="st">"pf"</span>,</span>
<span>  matriceDim.num     <span class="op">=</span> <span class="fl">41</span>,</span>
<span>  cores.num          <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose.bln        <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="case-2-interactions-around-genomic-regions-or-domains-">Case 2: Interactions around genomic regions or domains.<a class="anchor" aria-label="anchor" href="#case-2-interactions-around-genomic-regions-or-domains-"></a>
</h3>
<p>In this case, extracted matrices are resized and scaled in order to
fit all regions to the same number of bins.</p>
<p><img src="images/Extractions_of_Regions.png"><!-- --></p>
<div class="section level4">
<h4 id="regions-defined-with-ginteraction-object-or-pairs-granges">Regions defined with GInteraction object or Pairs GRanges<a class="anchor" aria-label="anchor" href="#regions-defined-with-ginteraction-object-or-pairs-granges"></a>
</h4>
<p>The region’s start is defined by the center of the first element and
the region’s end by the center of the second element.<br><img src="images/Regions_GInteractions.png"><!-- --></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">interactions_RFmatrix_ctrl.lst</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn         <span class="op">=</span> <span class="va">interactions.gni</span>,</span>
<span>  hic.cmx_lst        <span class="op">=</span> <span class="va">HiC_ctrl.cmx_lst</span>,</span>
<span>  res.num            <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  referencePoint.chr <span class="op">=</span> <span class="st">"rf"</span>,</span>
<span>  matriceDim.num     <span class="op">=</span> <span class="fl">101</span>,</span>
<span>  cores.num          <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose.bln        <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="regions-defined-with-granges">Regions defined with GRanges<a class="anchor" aria-label="anchor" href="#regions-defined-with-granges"></a>
</h4>
<p>The regions are directly defined by the ranges of GRanges object.</p>
<p><img src="images/Regions_GRanges.png"><!-- --></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains_RFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn         <span class="op">=</span> <span class="va">domains.gnr</span>,</span>
<span>  hic.cmx_lst        <span class="op">=</span> <span class="va">HiC_ctrl.cmx_lst</span>,</span>
<span>  referencePoint.chr <span class="op">=</span> <span class="st">"rf"</span>,</span>
<span>  matriceDim.num     <span class="op">=</span> <span class="fl">101</span>,</span>
<span>  cores.num          <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose.bln        <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="case-3-interactions-along-the-diagonal-">Case 3: Interactions along the diagonal.<a class="anchor" aria-label="anchor" href="#case-3-interactions-along-the-diagonal-"></a>
</h3>
<p><img src="images/Extractions_of_Ponctuals_Interactions.png"><!-- --></p>
<div class="section level4">
<h4 id="example-with-tad">Example with TAD:<a class="anchor" aria-label="anchor" href="#example-with-tad"></a>
</h4>
<p>Step 1: generate a GRanges object of TAD boundaries by concatenating
starts and ends of TADs.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains_Border.gnr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">domains.gnr</span>, <span class="fl">1</span>, <span class="st">"start"</span><span class="op">)</span>,</span>
<span>    <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">resize</a></span><span class="op">(</span><span class="va">domains.gnr</span>, <span class="fl">1</span>,  <span class="st">"end"</span> <span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>

<p>Step 2: Filter and reduce TAD boundaries GRanges object according to
HiC resolution (binSize) + Store TAD names.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains_Border_Bin.gnr</span> <span class="op">&lt;-</span> <span class="fu">GenomicTK</span><span class="fu">::</span><span class="fu"><a href="https://nchanard.github.io/GenomicTK/reference/BinGRanges.html" class="external-link">BinGRanges</a></span><span class="op">(</span></span>
<span>  gRange.gnr  <span class="op">=</span> <span class="va">domains_Border.gnr</span>,</span>
<span>  binSize.num <span class="op">=</span> <span class="va">binSize.num</span>,</span>
<span>  verbose.bln <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">domains_Border_Bin.gnr</span><span class="op">$</span><span class="va">subname</span> <span class="op">&lt;-</span> <span class="va">domains_Border_Bin.gnr</span><span class="op">$</span><span class="va">name</span></span>
<span><span class="va">domains_Border_Bin.gnr</span><span class="op">$</span><span class="va">name</span>    <span class="op">&lt;-</span> <span class="va">domains_Border_Bin.gnr</span><span class="op">$</span><span class="va">bin</span></span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains_Border_Bin.gnr</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead><tr>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
strand
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
score
</th>
<th style="text-align:center;">
class
</th>
<th style="text-align:center;">
bin
</th>
<th style="text-align:center;">
subname
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
2L:74
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
2L:74
</td>
<td style="text-align:center;">
Tad_1
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
94001
</td>
<td style="text-align:center;">
95000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
2L:95
</td>
<td style="text-align:center;">
3, 8
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
2L:95
</td>
<td style="text-align:center;">
Tad_1, Tad_2
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
102001
</td>
<td style="text-align:center;">
103000
</td>
<td style="text-align:center;">
<ul><li></ul>
</td>
<td style="text-align:center;">
2L:103
</td>
<td style="text-align:center;">
8
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
2L:103
</td>
<td style="text-align:center;">
Tad_2, Tad_3
</td>
</tr>
</tbody>
</table></details><p>Now we have a GRanges object. And for the exemples below we will need
the same informations but in an GInteraction object.<br>
### ICI JE NE COMPPREND PAS CE QU ON FAIT,</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">domains_Border_Bin.gni</span> <span class="op">&lt;-</span> <span class="fu">InteractionSet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></span><span class="op">(</span><span class="va">domains_Border_Bin.gnr</span>,<span class="va">domains_Border_Bin.gnr</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="8">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
First
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="8">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Second
</div>
</th>
</tr>
<tr>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:right;">
name
</th>
<th style="text-align:center;">
score
</th>
<th style="text-align:center;">
class
</th>
<th style="text-align:center;">
bin
</th>
<th style="text-align:center;">
subname
</th>
<th style="text-align:center;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
score
</th>
<th style="text-align:center;">
class
</th>
<th style="text-align:center;">
bin
</th>
<th style="text-align:center;">
subname
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:right;">
2L:74
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
2L:74
</td>
<td style="text-align:center;">
Tad_1
</td>
<td style="text-align:center;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:center;">
2L:74
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
2L:74
</td>
<td style="text-align:center;">
Tad_1
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
94001
</td>
<td style="text-align:center;">
95000
</td>
<td style="text-align:right;">
2L:95
</td>
<td style="text-align:center;">
3, 8
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
2L:95
</td>
<td style="text-align:center;">
Tad_1, Tad_2
</td>
<td style="text-align:center;">
2L
</td>
<td style="text-align:center;">
94001
</td>
<td style="text-align:center;">
95000
</td>
<td style="text-align:center;">
2L:95
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
3, 8
</td>
<td style="text-align:center;">
2L:95
</td>
<td style="text-align:center;">
Tad_1, Tad_2
</td>
</tr>
<tr>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
102001
</td>
<td style="text-align:center;">
103000
</td>
<td style="text-align:right;">
2L:103
</td>
<td style="text-align:center;">
8
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
2L:103
</td>
<td style="text-align:center;">
Tad_2, Tad_3
</td>
<td style="text-align:center;">
2L
</td>
<td style="text-align:center;">
102001
</td>
<td style="text-align:center;">
103000
</td>
<td style="text-align:center;">
2L:103
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
8
</td>
<td style="text-align:center;">
2L:103
</td>
<td style="text-align:center;">
Tad_2, Tad_3
</td>
</tr>
</tbody>
</table></details>
</div>
<div class="section level4">
<h4 id="ponctual-interactions-defined-with-granges">Ponctual interactions defined with GRanges<a class="anchor" aria-label="anchor" href="#ponctual-interactions-defined-with-granges"></a>
</h4>
<p>Here the start and the end of each ranges are in a same bin.</p>
<p><img src="images/Ponctuals_Interactions_GRanges.png"><!-- --></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">border_PFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn         <span class="op">=</span> <span class="va">domains_Border_Bin.gnr</span>,</span>
<span>  hic.cmx_lst        <span class="op">=</span> <span class="va">HiC_ctrl.cmx_lst</span>,</span>
<span>  referencePoint.chr <span class="op">=</span> <span class="st">"pf"</span>,</span>
<span>  matriceDim.num     <span class="op">=</span> <span class="fl">101</span>,</span>
<span>  verbose.bln        <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="ponctual-interactions-defined-with-ginteractions">Ponctual interactions defined with GInteractions<a class="anchor" aria-label="anchor" href="#ponctual-interactions-defined-with-ginteractions"></a>
</h4>
<p>Here the first element (blue on scheme) is in the same bin as the
second (red on scheme).</p>
<p><img src="images/Ponctuals_Interactions_GInteractions.png"><!-- --></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">border_PFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn         <span class="op">=</span> <span class="va">domains_Border_Bin.gni</span>,</span>
<span>  hic.cmx_lst        <span class="op">=</span> <span class="va">HiC_ctrl.cmx_lst</span>,</span>
<span>  referencePoint.chr <span class="op">=</span> <span class="st">"pf"</span>,</span>
<span>  matriceDim.num     <span class="op">=</span> <span class="fl">101</span>,</span>
<span>  verbose.bln        <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="tips-2">Tips<a class="anchor" aria-label="anchor" href="#tips-2"></a>
</h3>
<ol style="list-style-type: decimal">
<li>If <code>res.num</code> is null, the function heritates the
resolution <code>hic.cmx_lst</code> attributes.</li>
<li>referencePoint.chr is automatically set as “pf” if every anchors and
baits are on the same bin (Case of Ex: Extraction of submatrices around
TAD boundaries).</li>
</ol>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="filtrations">Filtrations<a class="anchor" aria-label="anchor" href="#filtrations"></a>
</h2>
<p>The modularity of the workflow allows the user to filter
interactions, pairs or extracted submatrices at any step of the analysis
process. FilterInteractions() function takes as input a GInteraction
object or a list of submatrices, a list defining targets to filter and a
selection function defining how target elements will be filtered.</p>
<div class="section level3">
<h3 id="target-list-definition">Target list definition:<a class="anchor" aria-label="anchor" href="#target-list-definition"></a>
</h3>
<p>Target list must have a particular structure. First it is a named
list. Each name of each element must be the name of column in the
GInteraction that you want filter (or in the attributes “interactions”
of the matrices list you want filter. Then each element must be a
character list to match to this column or a function that will test each
row in the column and return a bolean.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">structureTarget.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  first_colname_of_GInteraction  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"value"</span><span class="op">)</span>,</span>
<span>  second_colname_of_GInteraction <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">eachElement</span><span class="op">)</span><span class="op">{</span><span class="va">min_th</span><span class="op">&lt;</span><span class="va">value</span> <span class="op">&amp;&amp;</span> <span class="va">value</span><span class="op">&lt;</span><span class="va">max_th</span><span class="op">}</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Interactions, pairs or extracted submatrices are filtered by metadata
elements inherited from GRanges objects used in SearchPairs(). Those
metadata are stored in the attributes of the submatrices list and are
accessible as follow:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<details><summary>
View
</summary><table class="table">
<thead>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="7">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Ranges
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="15">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Metadata
</div>
</th>
</tr>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Names
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
First
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Second
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="5">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Interaction
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="5">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Anchor
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="5">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Bait
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:right;">
seq
</th>
<th style="text-align:center;">
start
</th>
<th style="text-align:center;">
end
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
constraint
</th>
<th style="text-align:center;">
distance
</th>
<th style="text-align:center;">
orientation
</th>
<th style="text-align:center;">
submatrix.name
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
bin
</th>
<th style="text-align:center;">
Beaf.name
</th>
<th style="text-align:center;">
Beaf.score
</th>
<th style="text-align:center;">
Beaf.bln
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
bin
</th>
<th style="text-align:center;">
Tss.name
</th>
<th style="text-align:center;">
Tss.class
</th>
<th style="text-align:center;">
Tss.bln
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
2L:74_2L:77
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
73001
</td>
<td style="text-align:center;">
74000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
76001
</td>
<td style="text-align:center;">
77000
</td>
<td style="text-align:center;">
2L:74_2L:77
</td>
<td style="text-align:center;">
Tad_1
</td>
<td style="text-align:center;">
3000
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
2L:74_2L:77
</td>
<td style="text-align:center;">
2L:74:Tad_1
</td>
<td style="text-align:center;">
2L:74
</td>
<td style="text-align:center;">
Beaf32_8
</td>
<td style="text-align:center;">
205
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:77:Tad_1
</td>
<td style="text-align:center;">
2L:77
</td>
<td style="text-align:center;">
FBgn0031214
</td>
<td style="text-align:center;">
inactive
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:115_2L:107
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
114001
</td>
<td style="text-align:center;">
115000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
106001
</td>
<td style="text-align:center;">
107000
</td>
<td style="text-align:center;">
2L:115_2L:107
</td>
<td style="text-align:center;">
Tad_3
</td>
<td style="text-align:center;">
8000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:107_2L:115
</td>
<td style="text-align:center;">
2L:115:Tad_3
</td>
<td style="text-align:center;">
2L:115
</td>
<td style="text-align:center;">
Beaf32_14
</td>
<td style="text-align:center;">
1707
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:107:Tad_3
</td>
<td style="text-align:center;">
2L:107
</td>
<td style="text-align:center;">
FBgn0026787, FBgn0005278
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:121_2L:107
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
120001
</td>
<td style="text-align:center;">
121000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
106001
</td>
<td style="text-align:center;">
107000
</td>
<td style="text-align:center;">
2L:121_2L:107
</td>
<td style="text-align:center;">
Tad_3
</td>
<td style="text-align:center;">
14000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:107_2L:121
</td>
<td style="text-align:center;">
2L:121:Tad_3
</td>
<td style="text-align:center;">
2L:121
</td>
<td style="text-align:center;">
Beaf32_15
</td>
<td style="text-align:center;">
484
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:107:Tad_3
</td>
<td style="text-align:center;">
2L:107
</td>
<td style="text-align:center;">
FBgn0026787, FBgn0005278
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:121_2L:117
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
120001
</td>
<td style="text-align:center;">
121000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
116001
</td>
<td style="text-align:center;">
117000
</td>
<td style="text-align:center;">
2L:121_2L:117
</td>
<td style="text-align:center;">
Tad_3
</td>
<td style="text-align:center;">
4000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:117_2L:121
</td>
<td style="text-align:center;">
2L:121:Tad_3
</td>
<td style="text-align:center;">
2L:121
</td>
<td style="text-align:center;">
Beaf32_15
</td>
<td style="text-align:center;">
484
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:117:Tad_3
</td>
<td style="text-align:center;">
2L:117
</td>
<td style="text-align:center;">
FBgn0031219
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:274_2L:268
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
273001
</td>
<td style="text-align:center;">
274000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
267001
</td>
<td style="text-align:center;">
268000
</td>
<td style="text-align:center;">
2L:274_2L:268
</td>
<td style="text-align:center;">
Tad_10
</td>
<td style="text-align:center;">
6000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:268_2L:274
</td>
<td style="text-align:center;">
2L:274:Tad_10
</td>
<td style="text-align:center;">
2L:274
</td>
<td style="text-align:center;">
Beaf32_39
</td>
<td style="text-align:center;">
210
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:268:Tad_10
</td>
<td style="text-align:center;">
2L:268
</td>
<td style="text-align:center;">
FBgn0031239
</td>
<td style="text-align:center;">
inactive
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:278_2L:268
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
277001
</td>
<td style="text-align:center;">
278000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
267001
</td>
<td style="text-align:center;">
268000
</td>
<td style="text-align:center;">
2L:278_2L:268
</td>
<td style="text-align:center;">
Tad_10
</td>
<td style="text-align:center;">
10000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:268_2L:278
</td>
<td style="text-align:center;">
2L:278:Tad_10
</td>
<td style="text-align:center;">
2L:278
</td>
<td style="text-align:center;">
Beaf32_41
</td>
<td style="text-align:center;">
1641
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:268:Tad_10
</td>
<td style="text-align:center;">
2L:268
</td>
<td style="text-align:center;">
FBgn0031239
</td>
<td style="text-align:center;">
inactive
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:274_2L:279
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
273001
</td>
<td style="text-align:center;">
274000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
278001
</td>
<td style="text-align:center;">
279000
</td>
<td style="text-align:center;">
2L:274_2L:279
</td>
<td style="text-align:center;">
Tad_10
</td>
<td style="text-align:center;">
5000
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
2L:274_2L:279
</td>
<td style="text-align:center;">
2L:274:Tad_10
</td>
<td style="text-align:center;">
2L:274
</td>
<td style="text-align:center;">
Beaf32_39
</td>
<td style="text-align:center;">
210
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:279:Tad_10
</td>
<td style="text-align:center;">
2L:279
</td>
<td style="text-align:center;">
FBgn0086855
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:306_2L:320
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
305001
</td>
<td style="text-align:center;">
306000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
319001
</td>
<td style="text-align:center;">
320000
</td>
<td style="text-align:center;">
2L:306_2L:320
</td>
<td style="text-align:center;">
Tad_13
</td>
<td style="text-align:center;">
14000
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
2L:306_2L:320
</td>
<td style="text-align:center;">
2L:306:Tad_13
</td>
<td style="text-align:center;">
2L:306
</td>
<td style="text-align:center;">
Beaf32_48
</td>
<td style="text-align:center;">
52
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:320:Tad_13
</td>
<td style="text-align:center;">
2L:320
</td>
<td style="text-align:center;">
FBgn0031248
</td>
<td style="text-align:center;">
inactive
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:471_2L:422
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
470001
</td>
<td style="text-align:center;">
471000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
421001
</td>
<td style="text-align:center;">
422000
</td>
<td style="text-align:center;">
2L:471_2L:422
</td>
<td style="text-align:center;">
Tad_17
</td>
<td style="text-align:center;">
49000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:422_2L:471
</td>
<td style="text-align:center;">
2L:471:Tad_17
</td>
<td style="text-align:center;">
2L:471
</td>
<td style="text-align:center;">
Beaf32_62
</td>
<td style="text-align:center;">
37
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:422:Tad_17
</td>
<td style="text-align:center;">
2L:422
</td>
<td style="text-align:center;">
FBgn0031253
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
2L:472_2L:422
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
471001
</td>
<td style="text-align:center;">
472000
</td>
<td style="text-align:right;">
2L
</td>
<td style="text-align:center;">
421001
</td>
<td style="text-align:center;">
422000
</td>
<td style="text-align:center;">
2L:472_2L:422
</td>
<td style="text-align:center;">
Tad_17
</td>
<td style="text-align:center;">
50000
</td>
<td style="text-align:center;">
FALSE
</td>
<td style="text-align:center;">
2L:422_2L:472
</td>
<td style="text-align:center;">
2L:472:Tad_17
</td>
<td style="text-align:center;">
2L:472
</td>
<td style="text-align:center;">
Beaf32_62
</td>
<td style="text-align:center;">
37
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2L:422:Tad_17
</td>
<td style="text-align:center;">
2L:422
</td>
<td style="text-align:center;">
FBgn0031253
</td>
<td style="text-align:center;">
active
</td>
<td style="text-align:center;">
1
</td>
</tr>
</tbody>
</table></details><div class="section level4">
<h4 id="example-of-target-list">Example of target list:<a class="anchor" aria-label="anchor" href="#example-of-target-list"></a>
</h4>
<p>In this example, Pairs will be filtered on anchor.Beaf.name,
bait.Tss.name, name (which correpond to submatrix IDs) and distance. The
aim of the example is to filter Pairs or submatrices that have: 1.
“Beaf32_8” and “Beaf32_15” in <strong>anchor.Beaf.name</strong> 2.
“FBgn0031214” and “FBgn0005278” in <strong>bait.Tss.name</strong><br>
3. <strong>distance</strong> exactly equal to 14000 or 3000<br>
And to exclude Pairs or submatrices that have: 4. “2L:74_2L:77” in
<strong>name</strong></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  anchor.Beaf.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Beaf32_8"</span>,<span class="st">"Beaf32_15"</span><span class="op">)</span>,</span>
<span>  bait.Tss.name    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FBgn0031214"</span>,<span class="st">"FBgn0005278"</span><span class="op">)</span>,</span>
<span>  name             <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2L:74_2L:77"</span><span class="op">)</span>,</span>
<span>  distance         <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">columnElement</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fl">14000</span><span class="op">==</span><span class="va">columnElement</span> <span class="op">||</span> <span class="va">columnElement</span> <span class="op">==</span> <span class="fl">3000</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="selection-function-definition">Selection Function definition:<a class="anchor" aria-label="anchor" href="#selection-function-definition"></a>
</h3>
<p>The selection function defines which operations (union(),
intersect(), setdiff()…) are done to filter the set of Pairs with target
elements. For more operation examples, see <em>Selection function tips
and examples</em> section.</p>
<div class="section level4">
<h4 id="example-of-selection-function-according-to-the-example-target">Example of selection function according to the example target<a class="anchor" aria-label="anchor" href="#example-of-selection-function-according-to-the-example-target"></a>
</h4>
<p>Following the example case defined in target.lst</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selection.fun</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">anchor.Beaf.name</span>, <span class="va">bait.Tss.name</span> ,<span class="va">distance</span><span class="op">)</span> <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="filtration-with-selection">Filtration with selection<a class="anchor" aria-label="anchor" href="#filtration-with-selection"></a>
</h3>
<div class="section level4">
<h4 id="example-of-ginteraction-object-filtration">Example of GInteraction object filtration<a class="anchor" aria-label="anchor" href="#example-of-ginteraction-object-filtration"></a>
</h4>
<p>With a GInteraction object as input, FilterInteractions() will return
the indices of filtered elements.</p>
<p>With the target.lst and selection.fun defined above:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  interarctions.gni <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span>,</span>
<span>  target.lst        <span class="op">=</span> <span class="va">target.lst</span>,</span>
<span>  selection.fun     <span class="op">=</span> <span class="va">selection.fun</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  int 3</span></span></code></pre></div>
<p>names of 4th interaction is:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] "2L:121_2L:117"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-of-matrices-list-filtration">Example of Matrices list filtration<a class="anchor" aria-label="anchor" href="#example-of-matrices-list-filtration"></a>
</h4>
<p>With a matrices list as input, FilterInteractions() will return the
filtered matrices list, with updated attributes.</p>
<p>With the target.lst and selection.fun defined above:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtred_interactions_RFmatrix_ctrl.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  target.lst    <span class="op">=</span> <span class="va">target.lst</span>,</span>
<span>  selection.fun <span class="op">=</span> <span class="va">selection.fun</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="specific-case-1-only-one-target-and-no-selection">Specific case 1: Only one target and no selection<a class="anchor" aria-label="anchor" href="#specific-case-1-only-one-target-and-no-selection"></a>
</h3>
<p>For example, we want to filter the top 100 first elements, so we
target the 100 first names</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">first100_target.lst</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  submatrix.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="ginteraction-filtration">GInteraction filtration<a class="anchor" aria-label="anchor" href="#ginteraction-filtration"></a>
</h4>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  interarctions.gni <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span>,</span>
<span>  target.lst        <span class="op">=</span> <span class="va">first100_target.lst</span>,</span>
<span>  selection.fun     <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  Named int [1:100] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">#&gt;  - attr(*, "names")= chr [1:100] "submatrix.name1" "submatrix.name2" "submatrix.name3" "submatrix.name4" ...</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="matrices-list-filtration">Matrices list filtration<a class="anchor" aria-label="anchor" href="#matrices-list-filtration"></a>
</h4>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">first100_interactions_RFmatrix_ctrl.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  target.lst    <span class="op">=</span> <span class="va">first100_target.lst</span>,</span>
<span>  selection.fun <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">first100_interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span></span>
<span><span class="co">#&gt; GInteractions object with 100 interactions and 15 metadata columns:</span></span>
<span><span class="co">#&gt;                     seqnames1           ranges1     seqnames2           ranges2</span></span>
<span><span class="co">#&gt;                         &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt;</span></span>
<span><span class="co">#&gt;         2L:74_2L:77        2L       73001-74000 ---        2L       76001-77000</span></span>
<span><span class="co">#&gt;       2L:115_2L:107        2L     114001-115000 ---        2L     106001-107000</span></span>
<span><span class="co">#&gt;       2L:121_2L:107        2L     120001-121000 ---        2L     106001-107000</span></span>
<span><span class="co">#&gt;       2L:121_2L:117        2L     120001-121000 ---        2L     116001-117000</span></span>
<span><span class="co">#&gt;       2L:274_2L:268        2L     273001-274000 ---        2L     267001-268000</span></span>
<span><span class="co">#&gt;                 ...       ...               ... ...       ...               ...</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12498        2L 12514001-12515000 ---        2L 12497001-12498000</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12498        2L 12515001-12516000 ---        2L 12497001-12498000</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12508        2L 12514001-12515000 ---        2L 12507001-12508000</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12508        2L 12515001-12516000 ---        2L 12507001-12508000</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12509        2L 12514001-12515000 ---        2L 12508001-12509000</span></span>
<span><span class="co">#&gt;                     |              name  constraint  distance orientation</span></span>
<span><span class="co">#&gt;                     |       &lt;character&gt; &lt;character&gt; &lt;integer&gt;   &lt;logical&gt;</span></span>
<span><span class="co">#&gt;         2L:74_2L:77 |       2L:74_2L:77       Tad_1      3000        TRUE</span></span>
<span><span class="co">#&gt;       2L:115_2L:107 |     2L:115_2L:107       Tad_3      8000       FALSE</span></span>
<span><span class="co">#&gt;       2L:121_2L:107 |     2L:121_2L:107       Tad_3     14000       FALSE</span></span>
<span><span class="co">#&gt;       2L:121_2L:117 |     2L:121_2L:117       Tad_3      4000       FALSE</span></span>
<span><span class="co">#&gt;       2L:274_2L:268 |     2L:274_2L:268      Tad_10      6000       FALSE</span></span>
<span><span class="co">#&gt;                 ... .               ...         ...       ...         ...</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12498 | 2L:12515_2L:12498     Tad_310     17000       FALSE</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12498 | 2L:12516_2L:12498     Tad_310     18000       FALSE</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12508 | 2L:12515_2L:12508     Tad_310      7000       FALSE</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12508 | 2L:12516_2L:12508     Tad_310      8000       FALSE</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12509 | 2L:12515_2L:12509     Tad_310      6000       FALSE</span></span>
<span><span class="co">#&gt;                        submatrix.name  anchor.bin      anchor.name    bait.bin</span></span>
<span><span class="co">#&gt;                           &lt;character&gt; &lt;character&gt;      &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;         2L:74_2L:77       2L:74_2L:77       2L:74      2L:74:Tad_1       2L:77</span></span>
<span><span class="co">#&gt;       2L:115_2L:107     2L:107_2L:115      2L:115     2L:115:Tad_3      2L:107</span></span>
<span><span class="co">#&gt;       2L:121_2L:107     2L:107_2L:121      2L:121     2L:121:Tad_3      2L:107</span></span>
<span><span class="co">#&gt;       2L:121_2L:117     2L:117_2L:121      2L:121     2L:121:Tad_3      2L:117</span></span>
<span><span class="co">#&gt;       2L:274_2L:268     2L:268_2L:274      2L:274    2L:274:Tad_10      2L:268</span></span>
<span><span class="co">#&gt;                 ...               ...         ...              ...         ...</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12498 2L:12498_2L:12515    2L:12515 2L:12515:Tad_310    2L:12498</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12498 2L:12498_2L:12516    2L:12516 2L:12516:Tad_310    2L:12498</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12508 2L:12508_2L:12515    2L:12515 2L:12515:Tad_310    2L:12508</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12508 2L:12508_2L:12516    2L:12516 2L:12516:Tad_310    2L:12508</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12509 2L:12509_2L:12515    2L:12515 2L:12515:Tad_310    2L:12509</span></span>
<span><span class="co">#&gt;                            bait.name anchor.Beaf.score anchor.Beaf.name</span></span>
<span><span class="co">#&gt;                          &lt;character&gt;         &lt;numeric&gt;           &lt;list&gt;</span></span>
<span><span class="co">#&gt;         2L:74_2L:77      2L:77:Tad_1               205         Beaf32_8</span></span>
<span><span class="co">#&gt;       2L:115_2L:107     2L:107:Tad_3              1707        Beaf32_14</span></span>
<span><span class="co">#&gt;       2L:121_2L:107     2L:107:Tad_3               484        Beaf32_15</span></span>
<span><span class="co">#&gt;       2L:121_2L:117     2L:117:Tad_3               484        Beaf32_15</span></span>
<span><span class="co">#&gt;       2L:274_2L:268    2L:268:Tad_10               210        Beaf32_39</span></span>
<span><span class="co">#&gt;                 ...              ...               ...              ...</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12498 2L:12498:Tad_310                44       Beaf32_900</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12498 2L:12498:Tad_310                88       Beaf32_901</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12508 2L:12508:Tad_310                44       Beaf32_900</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12508 2L:12508:Tad_310                88       Beaf32_901</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12509 2L:12509:Tad_310                44       Beaf32_900</span></span>
<span><span class="co">#&gt;                     anchor.Beaf.bln bait.Tss.class           bait.Tss.name</span></span>
<span><span class="co">#&gt;                           &lt;numeric&gt;    &lt;character&gt;                  &lt;list&gt;</span></span>
<span><span class="co">#&gt;         2L:74_2L:77               1       inactive             FBgn0031214</span></span>
<span><span class="co">#&gt;       2L:115_2L:107               1         active FBgn0026787,FBgn0005278</span></span>
<span><span class="co">#&gt;       2L:121_2L:107               1         active FBgn0026787,FBgn0005278</span></span>
<span><span class="co">#&gt;       2L:121_2L:117               1         active             FBgn0031219</span></span>
<span><span class="co">#&gt;       2L:274_2L:268               1       inactive             FBgn0031239</span></span>
<span><span class="co">#&gt;                 ...             ...            ...                     ...</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12498               1       inactive             FBgn0051858</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12498               1       inactive             FBgn0051858</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12508               1       inactive             FBgn0262353</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12508               1       inactive             FBgn0262353</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12509               1       inactive             FBgn0263458</span></span>
<span><span class="co">#&gt;                     bait.Tss.bln</span></span>
<span><span class="co">#&gt;                        &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;         2L:74_2L:77            1</span></span>
<span><span class="co">#&gt;       2L:115_2L:107            1</span></span>
<span><span class="co">#&gt;       2L:121_2L:107            1</span></span>
<span><span class="co">#&gt;       2L:121_2L:117            1</span></span>
<span><span class="co">#&gt;       2L:274_2L:268            1</span></span>
<span><span class="co">#&gt;                 ...          ...</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12498            1</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12498            1</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12508            1</span></span>
<span><span class="co">#&gt;   2L:12516_2L:12508            1</span></span>
<span><span class="co">#&gt;   2L:12515_2L:12509            1</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 429 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome</span></span></code></pre></div>
<p>A selection of some matrices removes attributes.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="specific-case-2-sampling">Specific case 2: Sampling<a class="anchor" aria-label="anchor" href="#specific-case-2-sampling"></a>
</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nSample.num</span> <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">target.lst</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span><span class="op">$</span><span class="va">name</span>,<span class="va">nSample.num</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="ginteraction-sampling">GInteraction sampling<a class="anchor" aria-label="anchor" href="#ginteraction-sampling"></a>
</h4>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  interarctions.gni <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span>,</span>
<span>  target.lst        <span class="op">=</span> <span class="va">target.lst</span>,</span>
<span>  selection.fun     <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  Named int [1:3] 14 179 195</span></span>
<span><span class="co">#&gt;  - attr(*, "names")= chr [1:3] "name1" "name2" "name3"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="matrices-list-sampling">Matrices list sampling<a class="anchor" aria-label="anchor" href="#matrices-list-sampling"></a>
</h4>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sampled_interactions_RFmatrix_ctrl.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  target.lst    <span class="op">=</span> <span class="va">target.lst</span>,</span>
<span>  selection.fun <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">sampled_interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span></span>
<span><span class="co">#&gt; GInteractions object with 3 interactions and 15 metadata columns:</span></span>
<span><span class="co">#&gt;                     seqnames1           ranges1     seqnames2           ranges2</span></span>
<span><span class="co">#&gt;                         &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt;</span></span>
<span><span class="co">#&gt;     2L:1178_2L:1174        2L   1177001-1178000 ---        2L   1173001-1174000</span></span>
<span><span class="co">#&gt;   2L:21574_2L:21492        2L 21573001-21574000 ---        2L 21491001-21492000</span></span>
<span><span class="co">#&gt;     2R:5672_2R:5653        2R   5671001-5672000 ---        2R   5652001-5653000</span></span>
<span><span class="co">#&gt;                     |              name  constraint  distance orientation</span></span>
<span><span class="co">#&gt;                     |       &lt;character&gt; &lt;character&gt; &lt;integer&gt;   &lt;logical&gt;</span></span>
<span><span class="co">#&gt;     2L:1178_2L:1174 |   2L:1178_2L:1174      Tad_35      4000       FALSE</span></span>
<span><span class="co">#&gt;   2L:21574_2L:21492 | 2L:21574_2L:21492     Tad_486     82000       FALSE</span></span>
<span><span class="co">#&gt;     2R:5672_2R:5653 |   2R:5672_2R:5653     Tad_513     19000       FALSE</span></span>
<span><span class="co">#&gt;                        submatrix.name  anchor.bin      anchor.name    bait.bin</span></span>
<span><span class="co">#&gt;                           &lt;character&gt; &lt;character&gt;      &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;     2L:1178_2L:1174   2L:1174_2L:1178     2L:1178   2L:1178:Tad_35     2L:1174</span></span>
<span><span class="co">#&gt;   2L:21574_2L:21492 2L:21492_2L:21574    2L:21574 2L:21574:Tad_486    2L:21492</span></span>
<span><span class="co">#&gt;     2R:5672_2R:5653   2R:5653_2R:5672     2R:5672  2R:5672:Tad_513     2R:5653</span></span>
<span><span class="co">#&gt;                            bait.name anchor.Beaf.score anchor.Beaf.name</span></span>
<span><span class="co">#&gt;                          &lt;character&gt;         &lt;numeric&gt;           &lt;list&gt;</span></span>
<span><span class="co">#&gt;     2L:1178_2L:1174   2L:1174:Tad_35               921       Beaf32_130</span></span>
<span><span class="co">#&gt;   2L:21574_2L:21492 2L:21492:Tad_486               529      Beaf32_1349</span></span>
<span><span class="co">#&gt;     2R:5672_2R:5653  2R:5653:Tad_513                99      Beaf32_1462</span></span>
<span><span class="co">#&gt;                     anchor.Beaf.bln bait.Tss.class bait.Tss.name bait.Tss.bln</span></span>
<span><span class="co">#&gt;                           &lt;numeric&gt;    &lt;character&gt;        &lt;list&gt;    &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;     2L:1178_2L:1174               1         active   FBgn0031318            1</span></span>
<span><span class="co">#&gt;   2L:21574_2L:21492               1       inactive   FBgn0053837            1</span></span>
<span><span class="co">#&gt;     2R:5672_2R:5653               1         active   FBgn0033028            1</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 429 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="specific-case-3-filterinteractions-without-selection-function">Specific case 3: FilterInteractions() without selection
function<a class="anchor" aria-label="anchor" href="#specific-case-3-filterinteractions-without-selection-function"></a>
</h3>
<p>Without any selection function, FilterInteractions() will return all
indices corresponding to each target in the list. Then, the indices of
interest can be selected in a second step.</p>
<div class="section level4">
<h4 id="ginteraction-filtration-1">GInteraction filtration<a class="anchor" aria-label="anchor" href="#ginteraction-filtration-1"></a>
</h4>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  interarctions.gni <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span>,</span>
<span>  target.lst        <span class="op">=</span> <span class="va">target.lst</span>,</span>
<span>  selection.fun     <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  Named int [1:3] 14 179 195</span></span>
<span><span class="co">#&gt;  - attr(*, "names")= chr [1:3] "name1" "name2" "name3"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="matrices-list-filtration-1">Matrices list filtration<a class="anchor" aria-label="anchor" href="#matrices-list-filtration-1"></a>
</h4>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  matrices.lst      <span class="op">=</span> <span class="va">interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  target.lst        <span class="op">=</span> <span class="va">target.lst</span>,</span>
<span>  selection.fun     <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 3</span></span>
<span><span class="co">#&gt;  $ 2L:1174_2L:1178  : num [1:101, 1:101] 0 NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ 2L:21492_2L:21574: num [1:101, 1:101] 0 NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ 2R:5653_2R:5672  : num [1:101, 1:101] 1.77 NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  - attr(*, "interactions")=Formal class 'GInteractions' [package "InteractionSet"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. ..@ anchor1        : int [1:3] 20 231 243</span></span>
<span><span class="co">#&gt;   .. ..@ anchor2        : int [1:3] 19 226 240</span></span>
<span><span class="co">#&gt;   .. ..@ regions        :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ values         : Factor w/ 2 levels "2L","2R": 1 2</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ lengths        : int [1:2] 236 193</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ start          : int [1:429] 73001 74001 76001 106001 114001 116001 120001 267001 273001 277001 ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ width          : int [1:429] 1000 1000 1000 1000 1000 1000 1000 1000 1000 1000 ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ NAMES          : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": 3</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ lengths        : int 429</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ seqnames   : chr [1:2] "2L" "2R"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ seqlengths : int [1:2] 23513712 25286936</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ is_circular: logi [1:2] NA NA</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ genome     : chr [1:2] NA NA</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ nrows          : int 429</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. ..@ NAMES          : chr [1:3] "2L:1178_2L:1174" "2L:21574_2L:21492" "2R:5672_2R:5653"</span></span>
<span><span class="co">#&gt;   .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ nrows          : int 3</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ listData       :List of 15</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ name             : chr [1:3] "2L:1178_2L:1174" "2L:21574_2L:21492" "2R:5672_2R:5653"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ constraint       : chr [1:3] "Tad_35" "Tad_486" "Tad_513"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ distance         : int [1:3] 4000 82000 19000</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ orientation      : logi [1:3] FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ submatrix.name   : chr [1:3] "2L:1174_2L:1178" "2L:21492_2L:21574" "2R:5653_2R:5672"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ anchor.bin       : chr [1:3] "2L:1178" "2L:21574" "2R:5672"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ anchor.name      : chr [1:3] "2L:1178:Tad_35" "2L:21574:Tad_486" "2R:5672:Tad_513"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ bait.bin         : chr [1:3] "2L:1174" "2L:21492" "2R:5653"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ bait.name        : chr [1:3] "2L:1174:Tad_35" "2L:21492:Tad_486" "2R:5653:Tad_513"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ anchor.Beaf.score: num [1:3] 921 529 99</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ anchor.Beaf.name :List of 3</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..$ : chr "Beaf32_130"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..$ : chr "Beaf32_1349"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..$ : chr "Beaf32_1462"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ anchor.Beaf.bln  : num [1:3] 1 1 1</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ bait.Tss.class   : chr [1:3] "active" "inactive" "active"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ bait.Tss.name    :List of 3</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..$ : chr "FBgn0031318"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..$ : chr "FBgn0053837"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..$ : chr "FBgn0033028"</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ bait.Tss.bln     : num [1:3] 1 1 1</span></span>
<span><span class="co">#&gt;   .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;  - attr(*, "target")=List of 1</span></span>
<span><span class="co">#&gt;   ..$ name: chr [1:3] "2L:21574_2L:21492" "2L:1178_2L:1174" "2R:5672_2R:5653"</span></span>
<span><span class="co">#&gt;  - attr(*, "resolution")= num 1000</span></span>
<span><span class="co">#&gt;  - attr(*, "referencePoint")= chr "rf"</span></span>
<span><span class="co">#&gt;  - attr(*, "matriceDim")= num 101</span></span>
<span><span class="co">#&gt;  - attr(*, "shiftFactor")= num 1</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="tips-3">Tips<a class="anchor" aria-label="anchor" href="#tips-3"></a>
</h3>
<ol style="list-style-type: decimal">
<li>Filter a GInterction object allows to cross the selected
index.<br>
</li>
<li>Filter a matrices list without selection is better than filter the
interaction attributes of the matrices list</li>
</ol>
<div class="section level4">
<h4 id="selection-function-tips-and-examples">Selection function tips and examples:<a class="anchor" aria-label="anchor" href="#selection-function-tips-and-examples"></a>
</h4>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"D"</span>,<span class="st">"G"</span><span class="op">)</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"E"</span>,<span class="st">"B"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span></span>
<span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"F"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Which is common to A, B and C</li>
</ol>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span>,<span class="va">c</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "G"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "G"</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Which is in A and/or B and/or C</li>
</ol>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">union</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span>,<span class="va">c</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A" "B" "C" "D" "E" "F" "G"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A" "B" "C" "D" "E" "F" "G"</span></span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Which is only in A</li>
</ol>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">setdiff</span>,<span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span>,<span class="va">c</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "D"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "D"</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Which is common in A with B, and not in C</li>
</ol>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "B"</span></span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>What is common in A with B, plus all that is present in C</li>
</ol>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A" "B" "C" "F" "G"</span></span></code></pre></div>
<ol start="6" style="list-style-type: decimal">
<li>What is common in C with all elements present in A and B</li>
</ol>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A" "C" "G"</span></span></code></pre></div>
<ol start="7" style="list-style-type: decimal">
<li>Everything that is present in A and B but not in C</li>
</ol>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "B" "D" "E"</span></span></code></pre></div>
<ol start="8" style="list-style-type: decimal">
<li>Which is present only once</li>
</ol>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">a</span>,<span class="va">b</span>,<span class="va">c</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">d</span>,<span class="va">d</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "D" "E" "F"</span></span></code></pre></div>
<hr>
</div>
</div>
</div>
<div class="section level2">
<h2 id="orientation">Orientation<a class="anchor" aria-label="anchor" href="#orientation"></a>
</h2>
<p>ExtractSubmatrix() returns submatrices orientated according to
5’-&gt;3’ orientation of the chromosome. In the case of heterotypic or
asymetric pairs (anchor != bait), anchors and baits are thus mixed on Y
and X axis of the matrices.</p>
<p><img src="images/Orientation_extraction.png"><!-- --></p>
<p>OrienteMatrix() function allows to force all matrices to be
orientated in a way that anchors will be systematically on Y axis and
baits on X axis.</p>
<p><img src="images/Orientation.png"><!-- --></p>
<div class="section level3">
<h3 id="acces-orientation-information">Acces orientation information<a class="anchor" aria-label="anchor" href="#acces-orientation-information"></a>
</h3>
<p>First, lets look how two matrices are oriented (column orientation of
metadata of the interactions attribute). Below we can see that the first
matrix is correctly oriented (value is TRUE) but not the second one.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">first100_interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">$</span><span class="va">interactions</span><span class="op">$</span><span class="va">orientation</span><span class="op">[</span><span class="fl">13</span><span class="op">:</span><span class="fl">14</span><span class="op">]</span> </span>
<span><span class="co">#&gt; [1]  TRUE FALSE</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>The 13th matrice is well oriented, i.e. the <strong>anchor Beaf is
in Y axis</strong> and the <strong>bait TSS in X axis</strong><br>
</li>
<li>The 14th matrice is not well oriented, i.e. the <strong>bait TSS is
in Y axis</strong> and the <strong>anchor Beaf in X axis</strong>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="orientation-on-matrices-list">Orientation on matrices list<a class="anchor" aria-label="anchor" href="#orientation-on-matrices-list"></a>
</h3>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OrienteMatrix.html">OrienteMatrix</a></span><span class="op">(</span><span class="va">first100_interactions_RFmatrix_ctrl.lst</span><span class="op">)</span></span></code></pre></div>
<!-- 
# I think we don't need to porve in the tuto that the orientation function works well...
### First matrice (don't need orientation)  

1. upper left corner **before orientation**.  


```r
first100_interactions_RFmatrix_ctrl.lst[[13]][1:5,1:5]
```

<details>  
<summary>View</summary>  
<table>
 <thead>
  <tr>
   <th style="text-align:center;"> 1 </th>
   <th style="text-align:center;"> 2 </th>
   <th style="text-align:center;"> 3 </th>
   <th style="text-align:center;"> 4 </th>
   <th style="text-align:center;"> 5 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.1 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.1 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.1 </td>
  </tr>
</tbody>
</table>
</details>  

  
2. upper left corner **after orientation does not change**.  


```r
oriented_first100_interactions_RFmatrix_ctrl.lst[[13]][1:5,1:5]
```

<details>  
<summary>View</summary>  
<table>
 <thead>
  <tr>
   <th style="text-align:center;"> 1 </th>
   <th style="text-align:center;"> 2 </th>
   <th style="text-align:center;"> 3 </th>
   <th style="text-align:center;"> 4 </th>
   <th style="text-align:center;"> 5 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.1 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.1 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.1 </td>
  </tr>
</tbody>
</table>
</details>  

### Second matrice (need orientation)  

1. upper left corner **before orientation**  


```r
first100_interactions_RFmatrix_ctrl.lst[[14]][1:5,1:5]
```

<details>  
<summary>View</summary>  
<table>
 <thead>
  <tr>
   <th style="text-align:center;"> 1 </th>
   <th style="text-align:center;"> 2 </th>
   <th style="text-align:center;"> 3 </th>
   <th style="text-align:center;"> 4 </th>
   <th style="text-align:center;"> 5 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0.2 </td>
   <td style="text-align:center;"> 0.4 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.7 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.2 </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.6 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.4 </td>
   <td style="text-align:center;"> 0.5 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.5 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.4 </td>
  </tr>
</tbody>
</table>
</details>  

2. **transposed lower right corner after orientation correspond to the upper left corner before orientation**  


```r
oriented_first100_interactions_RFmatrix_ctrl.lst[[14]][97:101,97:101]
```

<details>  
<summary>View</summary>  
<table>
 <thead>
  <tr>
   <th style="text-align:center;"> 1 </th>
   <th style="text-align:center;"> 2 </th>
   <th style="text-align:center;"> 3 </th>
   <th style="text-align:center;"> 4 </th>
   <th style="text-align:center;"> 5 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:center;"> 0.4 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.6 </td>
   <td style="text-align:center;"> 0.7 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.4 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.5 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.4 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.2 </td>
   <td style="text-align:center;"> 0.2 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.0 </td>
  </tr>
</tbody>
</table>
</details>  


## Orientation on only one matrice.
The function could also oriente only one matrix but in this case there is no attribute verification so you can oriente a matrix that don't need orentation.

### First matrix  

Transposed lower right corner after orientation correspond to the upper left corner before orientation.  


```r
OrienteMatrix(first100_interactions_RFmatrix_ctrl.lst[[13]])[97:101,97:101]
```

<details>  
<summary>View</summary>  
<table>
 <thead>
  <tr>
   <th style="text-align:center;"> 1 </th>
   <th style="text-align:center;"> 2 </th>
   <th style="text-align:center;"> 3 </th>
   <th style="text-align:center;"> 4 </th>
   <th style="text-align:center;"> 5 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:center;"> 0.1 </td>
   <td style="text-align:center;"> 0.1 </td>
   <td style="text-align:center;"> 0.1 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.0 </td>
   <td style="text-align:center;"> 0.0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.0 </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
   <td style="text-align:center;"> 0 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0 </td>
  </tr>
</tbody>
</table>
</details>  

### Second matrix
Below we find again the orientation of the second matrix.  


```r
OrienteMatrix(first100_interactions_RFmatrix_ctrl.lst[[14]])[97:101,97:101]
```

<details>  
<summary>View</summary>  
<table>
 <thead>
  <tr>
   <th style="text-align:center;"> 1 </th>
   <th style="text-align:center;"> 2 </th>
   <th style="text-align:center;"> 3 </th>
   <th style="text-align:center;"> 4 </th>
   <th style="text-align:center;"> 5 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:center;"> 0.4 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.6 </td>
   <td style="text-align:center;"> 0.7 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.4 </td>
   <td style="text-align:center;"> 0.5 </td>
   <td style="text-align:center;"> 0.5 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.3 </td>
   <td style="text-align:center;"> 0.4 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.2 </td>
   <td style="text-align:center;"> 0.2 </td>
  </tr>
  <tr>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> NA </td>
   <td style="text-align:center;"> 0.0 </td>
  </tr>
</tbody>
</table>
</details>   -->
<hr>
</div>
</div>
<div class="section level2">
<h2 id="quantifications">Quantifications<a class="anchor" aria-label="anchor" href="#quantifications"></a>
</h2>
<p>GetQuantif() function takes as input a list of submatrices and
returns a vector of contact frequencies on a given area, compute with a
given function:</p>
<div class="section level3">
<h3 id="basic-quantifications">Basic quantifications<a class="anchor" aria-label="anchor" href="#basic-quantifications"></a>
</h3>
<p>the <code>GetQuantif</code> function has some default area and
function (see <code><a href="../reference/GetQuantif.html">Genomic2DTK::GetQuantif</a></code>).</p>
<ol style="list-style-type: decimal">
<li>
<strong>area</strong>: Key word defining on which area the values
are extracted in each matrices.<br>
</li>
<li>
<strong>operation</strong> Key word defining which operation is done
on extracted values for each matrices.</li>
</ol>
<p>Example: Interactions values on the centered 3x3 square centred,
averaged after removing all zeros.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">center.num</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetQuantif.html">GetQuantif</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  area.fun      <span class="op">=</span> <span class="st">"center"</span>,</span>
<span>  operation.fun <span class="op">=</span> <span class="st">"mean"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<!-- 

Below we can see the values and the attributes of the numerical vector with a value for each matrices. There are same attributes than the matrices list with two news:  



```r
head(center.num)
#> $`2L:74_2L:77`
#> [1] 0.002752447
#> 
#> $`2L:115_2L:107`
#> [1] 0.004792452
#> 
#> $`2L:121_2L:107`
#> [1] 1.01102
#> 
#> $`2L:121_2L:117`
#> [1] 0.03643152
#> 
#> $`2L:274_2L:268`
#> [1] 1.69169
#> 
#> $`2L:278_2L:268`
#> [1] 1.483054
```

In order to print quantifications use `c` function to avoid the attributes printing. 

```r
head(c(center.num))
#> $`2L:74_2L:77`
#> [1] 0.002752447
#> 
#> $`2L:115_2L:107`
#> [1] 0.004792452
#> 
#> $`2L:121_2L:107`
#> [1] 1.01102
#> 
#> $`2L:121_2L:117`
#> [1] 0.03643152
#> 
#> $`2L:274_2L:268`
#> [1] 1.69169
#> 
#> $`2L:278_2L:268`
#> [1] 1.483054
```

And to reduce the printing use `unlist` function.  

```r
head(unlist(c(center.num)))
#>   2L:74_2L:77 2L:115_2L:107 2L:121_2L:107 2L:121_2L:117 2L:274_2L:268 
#>   0.002752447   0.004792452   1.011020060   0.036431519   1.691689943 
#> 2L:278_2L:268 
#>   1.483054107
```
   -->
</div>
<div class="section level3">
<h3 id="custom-functions">Custom functions<a class="anchor" aria-label="anchor" href="#custom-functions"></a>
</h3>
<p>The <code>GetQuantif</code> function also takes custom
<strong>area</strong> and <strong>operation</strong> in parameter.</p>
<ol style="list-style-type: decimal">
<li>
<strong>area</strong>: function defining on which submatrix
coordinates the values are extracted in each matrices.<br>
</li>
<li>
<strong>operation</strong> function defining which operation is done
on extracted values for each matrices.</li>
</ol>
<p>Example: Interactions values on the matrice.mtx[33:35,67:69] area,
averaged after removing all zeros.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/GetQuantif.html">GetQuantif</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  area.fun      <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">matrice.mtx</span><span class="op">)</span><span class="op">{</span><span class="va">matrice.mtx</span><span class="op">[</span><span class="fl">33</span><span class="op">:</span><span class="fl">35</span>,<span class="fl">67</span><span class="op">:</span><span class="fl">69</span><span class="op">]</span><span class="op">}</span>,</span>
<span>  operation.fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">area.mtx</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">area.mtx</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">area.mtx</span><span class="op">==</span><span class="fl">0</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="cn">NA</span>;</span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">area.mtx</span>,na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   2L:74_2L:77 2L:115_2L:107 2L:121_2L:107 2L:121_2L:117 2L:274_2L:268 </span></span>
<span><span class="co">#&gt;    0.01238601    0.04313207    1.01102006    0.03643152    1.69168994 </span></span>
<span><span class="co">#&gt; 2L:278_2L:268 </span></span>
<span><span class="co">#&gt;    1.48305411</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="particular-cases">Particular cases:<a class="anchor" aria-label="anchor" href="#particular-cases"></a>
</h3>
<div class="section level4">
<h4 id="values-naming">Values naming<a class="anchor" aria-label="anchor" href="#values-naming"></a>
</h4>
<p>By default, returned values are named with submatrix ID. If name.chr
is set with an element metadata column name from GInteraction
attributes, values are returned values are named according to this
element.</p>
<p>Example: Named quantifications with <code>anchor.Beaf.name</code></p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">namedCenter.num</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetQuantif.html">GetQuantif</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  area.fun      <span class="op">=</span> <span class="st">"center"</span>,</span>
<span>  operation.fun <span class="op">=</span> <span class="st">"mean"</span>,</span>
<span>  name.chr      <span class="op">=</span> <span class="st">"anchor.Beaf.name"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Note that changing submatrix ID for other names can create names
duplicates:</p>
Example: The 46th matrix is correspond to two Beaf32 peaks, i.e. it has
two anchor.Beaf.name
<table class="table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:center;">
name
</th>
<th style="text-align:center;">
anchor.Beaf.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
45
</td>
<td style="text-align:center;">
2L:5909_2L:5941
</td>
<td style="text-align:center;">
Beaf32_415
</td>
</tr>
<tr>
<td style="text-align:left;">
46
</td>
<td style="text-align:center;">
2L:5944_2L:5968
</td>
<td style="text-align:center;">
Beaf32_417, Beaf32_418
</td>
</tr>
<tr>
<td style="text-align:left;">
47
</td>
<td style="text-align:center;">
2L:5945_2L:5968
</td>
<td style="text-align:center;">
Beaf32_418
</td>
</tr>
<tr>
<td style="text-align:left;">
48
</td>
<td style="text-align:center;">
2L:6051_2L:6069
</td>
<td style="text-align:center;">
Beaf32_430
</td>
</tr>
<tr>
<td style="text-align:left;">
49
</td>
<td style="text-align:center;">
2L:6066_2L:6069
</td>
<td style="text-align:center;">
Beaf32_432
</td>
</tr>
<tr>
<td style="text-align:left;">
50
</td>
<td style="text-align:center;">
2L:6084_2L:6071
</td>
<td style="text-align:center;">
Beaf32_435
</td>
</tr>
</tbody>
</table>
<p>By consequence, the value in center.num is duplicated in
namedCenter.num</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">center.num</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">45</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span></span>
<span><span class="co">#&gt; 2L:5909_2L:5941 2L:5944_2L:5968 2L:5945_2L:5968 2L:6051_2L:6069 2L:6066_2L:6069 </span></span>
<span><span class="co">#&gt;       0.1871565       0.0000000       0.0000000       0.5279512       2.1392992 </span></span>
<span><span class="co">#&gt; 2L:6084_2L:6071 </span></span>
<span><span class="co">#&gt;       0.1096642</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">namedCenter.num</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">45</span><span class="op">:</span><span class="fl">51</span><span class="op">]</span></span>
<span><span class="co">#&gt; Beaf32_415 Beaf32_417 Beaf32_418 Beaf32_418 Beaf32_430 Beaf32_432 Beaf32_435 </span></span>
<span><span class="co">#&gt;  0.1871565  0.0000000  0.0000000  0.0000000  0.5279512  2.1392992  0.1096642</span></span></code></pre></div>
<p>Duplicated value index are stored in attributes.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">center.num</span><span class="op">)</span><span class="op">$</span><span class="va">duplicated</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">namedCenter.num</span><span class="op">)</span><span class="op">$</span><span class="va">duplicated</span></span>
<span><span class="co">#&gt; [1] 47</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="one-value-extraction">One value extraction<a class="anchor" aria-label="anchor" href="#one-value-extraction"></a>
</h4>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/GetQuantif.html">GetQuantif</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  area.fun      <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">matrice.mtx</span><span class="op">)</span><span class="op">{</span><span class="va">matrice.mtx</span><span class="op">[</span><span class="fl">5</span>,<span class="fl">5</span><span class="op">]</span><span class="op">}</span>,</span>
<span>  operation.fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">area.mtx</span><span class="op">)</span><span class="op">{</span><span class="va">area.mtx</span><span class="op">}</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   2L:74_2L:77 2L:115_2L:107 2L:121_2L:107 2L:121_2L:117 2L:274_2L:268 </span></span>
<span><span class="co">#&gt;     0.0000000     0.0000000     0.3403983     0.4351463     0.0000000 </span></span>
<span><span class="co">#&gt; 2L:278_2L:268 </span></span>
<span><span class="co">#&gt;     0.1806280</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="area-extraction">Area extraction<a class="anchor" aria-label="anchor" href="#area-extraction"></a>
</h4>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/GetQuantif.html">GetQuantif</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  area.fun      <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">matrice.mtx</span><span class="op">)</span><span class="op">{</span><span class="va">matrice.mtx</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span>,<span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">}</span>,</span>
<span>  operation.fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">area</span><span class="op">)</span><span class="op">{</span><span class="va">area</span><span class="op">}</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`2L:74_2L:77`</span></span>
<span><span class="co">#&gt;      [,1] [,2] [,3]</span></span>
<span><span class="co">#&gt; [1,]    0    0    0</span></span>
<span><span class="co">#&gt; [2,]   NA    0    0</span></span>
<span><span class="co">#&gt; [3,]   NA   NA    0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:115_2L:107`</span></span>
<span><span class="co">#&gt;      [,1] [,2]      [,3]</span></span>
<span><span class="co">#&gt; [1,]    0    0 0.1671591</span></span>
<span><span class="co">#&gt; [2,]   NA    0 0.2228788</span></span>
<span><span class="co">#&gt; [3,]   NA   NA 0.1857323</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:121_2L:107`</span></span>
<span><span class="co">#&gt;           [,1]      [,2]      [,3]</span></span>
<span><span class="co">#&gt; [1,] 0.3009771 0.7871710 1.2376943</span></span>
<span><span class="co">#&gt; [2,]        NA 0.3403983 0.5352191</span></span>
<span><span class="co">#&gt; [3,]        NA        NA 0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:121_2L:117`</span></span>
<span><span class="co">#&gt;           [,1]      [,2]      [,3]</span></span>
<span><span class="co">#&gt; [1,] 0.6591566 0.5355647 0.4119729</span></span>
<span><span class="co">#&gt; [2,]        NA 0.4351463 0.3347280</span></span>
<span><span class="co">#&gt; [3,]        NA        NA 0.2574830</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:274_2L:268`</span></span>
<span><span class="co">#&gt;      [,1] [,2] [,3]</span></span>
<span><span class="co">#&gt; [1,]    0    0    0</span></span>
<span><span class="co">#&gt; [2,]   NA    0    0</span></span>
<span><span class="co">#&gt; [3,]   NA   NA    0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:278_2L:268`</span></span>
<span><span class="co">#&gt;      [,1]      [,2]      [,3]</span></span>
<span><span class="co">#&gt; [1,]    0 0.2032066 0.5080164</span></span>
<span><span class="co">#&gt; [2,]   NA 0.1806280 0.4515701</span></span>
<span><span class="co">#&gt; [3,]   NA        NA 0.2822313</span></span></code></pre></div>
<p>If is null operation.fun return values of the selected area without
NA.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/GetQuantif.html">GetQuantif</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  area.fun      <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">matrice.mtx</span><span class="op">)</span><span class="op">{</span><span class="va">matrice.mtx</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span>,<span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">}</span>,</span>
<span>  operation.fun <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`2L:74_2L:77`</span></span>
<span><span class="co">#&gt; [1] 0 0 0 0 0 0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:115_2L:107`</span></span>
<span><span class="co">#&gt; [1] 0.0000000 0.0000000 0.0000000 0.1671591 0.2228788 0.1857323</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:121_2L:107`</span></span>
<span><span class="co">#&gt; [1] 0.3009771 0.7871710 0.3403983 1.2376943 0.5352191 0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:121_2L:117`</span></span>
<span><span class="co">#&gt; [1] 0.6591566 0.5355647 0.4351463 0.4119729 0.3347280 0.2574830</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:274_2L:268`</span></span>
<span><span class="co">#&gt; [1] 0 0 0 0 0 0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2L:278_2L:268`</span></span>
<span><span class="co">#&gt; [1] 0.0000000 0.2032066 0.1806280 0.5080164 0.4515701 0.2822313</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="tips-4">Tips<a class="anchor" aria-label="anchor" href="#tips-4"></a>
</h3>
<ol style="list-style-type: decimal">
<li>To print quantifications use <code>c</code> function to avoid the
attributes printing.<br>
</li>
<li>When you select area if is null <code>operation.fun</code> returns
values of the selected area without NA.</li>
</ol>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="aggregations">Aggregations<a class="anchor" aria-label="anchor" href="#aggregations"></a>
</h2>
<p>Aggregation() function takes as input a list of submatrices and
returns an aggregated matrix using a user defined aggregation
function.</p>
<div class="section level3">
<h3 id="one-sample-aggregation">One sample aggregation<a class="anchor" aria-label="anchor" href="#one-sample-aggregation"></a>
</h3>
<div class="section level4">
<h4 id="basic-aggregation">Basic aggregation<a class="anchor" aria-label="anchor" href="#basic-aggregation"></a>
</h4>
<p>Aggregation() function has some default aggregation functions like
<code>sum</code>, <code>mean</code> or <code>median</code> (see
<code><a href="../reference/Aggregation.html">Genomic2DTK::Aggregation</a></code>)</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">agg_sum.mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Aggregation.html">Aggregation</a></span><span class="op">(</span></span>
<span>  matrices.lst <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>, </span>
<span>  agg.fun      <span class="op">=</span> <span class="st">"sum"</span>,</span>
<span>  rm0.bln      <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<!-- 
Verification with extraction of one value  
ICI ENCORE, PAS BESON DE MONTRER que ton outils sait faire une somme...


```r
agg_sum.mtx[5,5]
#> [1] 21.63697
```



```r
GetQuantif(
  matrices.lst = oriented_first100_interactions_RFmatrix_ctrl.lst,
  area.fun     = function(matrice.mtx){matrice.mtx[5,5]},
  operation.fun = function(area.num){area.num}
  ) |> unlist() |> sum(na.rm=TRUE)
#> [1] 21.63697
```
 -->
</div>
<div class="section level4">
<h4 id="custom-aggregation">Custom aggregation<a class="anchor" aria-label="anchor" href="#custom-aggregation"></a>
</h4>
<p>Defining a custom aggregation function, example with mean function
removing NA:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">agg_mean.mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Aggregation.html">Aggregation</a></span><span class="op">(</span></span>
<span>  matrices.lst <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  agg.fun      <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x</span>,na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">}</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<!-- 

Verification with extraction of one value

```r
agg_mean.mtx[5,5]
#> [1] 0.2163697
```

```r
GetQuantif(
  matrices.lst = oriented_first100_interactions_RFmatrix_ctrl.lst,
  area.fun     = function(matrice.mtx){matrice.mtx[5,5]},
  operation.fun = function(area.num){area.num}
  ) |>
unlist() |>
mean(na.rm=TRUE)
#> [1] 0.2163697
```

 -->
<!-- 

### Distance constraint, mentioned in tips only
You could add distance constraint, in this case the function will agregate all matrices that have a distance between anchor and bait superior or equal to the `minDist` parameter and inferior or equal to the `maxDist` parameter

```r
agg_dist.mtx <- Aggregation(
  matrices.lst = oriented_first100_interactions_RFmatrix_ctrl.lst,
  minDist      = 9000,
  maxDist      = 11000,
  agg.fun      = "sum"
  )
```
Verification with extraction of one value = again, verification useless

```r
agg_dist.mtx[5,5]
#>      V97 
#> 2.674181
quantif.num <- GetQuantif(
  matrices.lst = oriented_first100_interactions_RFmatrix_ctrl.lst,
  area.fun     = function(matrice.mtx){matrice.mtx[5,5] },
  operation.fun = function(area.num){area.num}
  )
ndx <- which(
  attributes(quantif.num)$interactions$distance >= 9000 &
  attributes(quantif.num)$interactions$distance <= 11000
  )
sum(unlist(quantif.num)[ndx],na.rm=TRUE)
#> [1] 2.674181
```

 -->
</div>
</div>
<div class="section level3">
<h3 id="two-samples-differential-aggregation">Two samples differential aggregation<a class="anchor" aria-label="anchor" href="#two-samples-differential-aggregation"></a>
</h3>
<p>Aggregation() function can take as input two list of submatrices from
two samples or conditions and returns a differential aggregated matrix.
Two ways to obtain differential aggregation are applied, first is by
assessing differences on each individual pairs of matrices then
aggregate the differences; second is by aggregating matrices and assess
differences on aggregated matrices.</p>
<div class="section level4">
<h4 id="preparation-of-matrices-list">Preparation of matrices list<a class="anchor" aria-label="anchor" href="#preparation-of-matrices-list"></a>
</h4>
<ol style="list-style-type: decimal">
<li>Preparation of Control matrices list condition<br><strong>Filtration</strong>
</li>
</ol>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">first100_target.lst</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  submatrix.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">first100_interactions_RFmatrix_ctrl.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  target.lst    <span class="op">=</span> <span class="va">first100_target.lst</span>,</span>
<span>  selection.fun <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><strong>Orientation</strong></p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OrienteMatrix.html">OrienteMatrix</a></span><span class="op">(</span><span class="va">first100_interactions_RFmatrix_ctrl.lst</span><span class="op">)</span></span></code></pre></div>
<!-- 

**Quantifications for Verification**

```r
quantif_ctrl.num <- GetQuantif(
  matrices.lst  = oriented_first100_interactions_RFmatrix_ctrl.lst,
  area.fun      = function(matrice.mtx){matrice.mtx[5,5]},
  operation.fun = function(area.num){area.num}
  )
```
 -->
<ol start="2" style="list-style-type: decimal">
<li>Preparation of second matrices list in Beaf depleted condition.
<strong>Extraction</strong>
</li>
</ol>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">interactions_RFmatrix.lst</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractSubmatrix.html">ExtractSubmatrix</a></span><span class="op">(</span></span>
<span>  feature.gn         <span class="op">=</span> <span class="va">interactions.gni</span>,</span>
<span>  hic.cmx_lst        <span class="op">=</span> <span class="va">HiC.cmx_lst</span>,</span>
<span>  referencePoint.chr <span class="op">=</span> <span class="st">"rf"</span>,</span>
<span>  matriceDim.num     <span class="op">=</span> <span class="fl">101</span>,</span>
<span>  verbose            <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><strong>Filtration</strong></p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">first100_interactions_RFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterInteractions.html">FilterInteractions</a></span><span class="op">(</span></span>
<span>  matrices.lst  <span class="op">=</span> <span class="va">interactions_RFmatrix.lst</span>,</span>
<span>  target.lst    <span class="op">=</span> <span class="va">first100_target.lst</span>,</span>
<span>  selection.fun <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><strong>Orientation</strong></p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">oriented_first100_interactions_RFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OrienteMatrix.html">OrienteMatrix</a></span><span class="op">(</span><span class="va">first100_interactions_RFmatrix.lst</span><span class="op">)</span></span></code></pre></div>
<!-- 


**Quantifications for Verification**  

```r
quantif.num <- GetQuantif(
  matrices.lst  = oriented_first100_interactions_RFmatrix.lst,
  area.fun      = function(matrice.mtx){matrice.mtx[5,5]},
  operation.fun = function(area.num){area.num}
  )
```

 -->
</div>
<div class="section level4">
<h4 id="aggregate">Aggregate<a class="anchor" aria-label="anchor" href="#aggregate"></a>
</h4>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diffAggreg.mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Aggregation.html">Aggregation</a></span><span class="op">(</span></span>
<span>  ctrlMatrices.lst    <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>  matrices.lst        <span class="op">=</span> <span class="va">oriented_first100_interactions_RFmatrix.lst</span>,</span>
<span>  minDist             <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxDist             <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  agg.fun             <span class="op">=</span> <span class="st">"mean"</span>,</span>
<span>  rm0.bln             <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  diff.fun            <span class="op">=</span> <span class="st">"substraction"</span>,</span>
<span>  scaleCorrection.bln <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  correctionArea.lst  <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    i <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span>,</span>
<span>    j <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">72</span><span class="op">:</span><span class="fl">101</span><span class="op">)</span></span>
<span>    <span class="op">)</span>,</span>
<span>  statCompare.bln <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<!-- 
### Display Aggregation useless in tutorial, we don't need to validate we know how to do a substraction !

1. Print with all attributes  


```r
diffAggreg.mtx
```

2. Print only matrice  


```r
prmatrix(diffAggreg.mtx)
diffAggreg.mtx[1:101,1:101]
```

3. Differential at each submatrix level  


```r
diffAggreg.mtx[5,5]
#> [1] 0.03645583
mean(unlist(quantif.num) - unlist(quantif_ctrl.num))
#> [1] 0.03645583
```

4. Aggregation of Control  


```r
attributes(diffAggreg.mtx)$matrices$aggCtrl[5,5]
#> [1] 0.2163697
mean(unlist(quantif_ctrl.num))
#> [1] 0.2163697
```

5. Aggregation of Beaf depleted Condition  


```r
attributes(diffAggreg.mtx)$matrices$agg[5,5]
#> [1] 0.2528255
mean(unlist(quantif.num))
#> [1] 0.2528255
```

6. Differential at Aggregated level  


```r
attributes(diffAggreg.mtx)$matrices$agg[5,5]-attributes(diffAggreg.mtx)$matrices$aggCtrl[5,5]
#> [1] 0.03645583
attributes(diffAggreg.mtx)$matrices$aggDelta[5,5]
#> [1] 0.03645583
mean(unlist(quantif.num)) - mean(unlist(quantif_ctrl.num))
#> [1] 0.03645583
```

7. ScaleFactor  


```r
correctedFact.num = median(attributes(diffAggreg.mtx)$matrices$aggCtrl[1:30,72:101]) - median(attributes(diffAggreg.mtx)$matrices$agg[1:30,72:101])
correctedFact.num
#> [1] -0.002255729
attributes(diffAggreg.mtx)$correctedFact
#> [1] -0.002255729
```

8. Aggregation Classic Scaled  


```r
attributes(diffAggreg.mtx)$matrices$aggCorrected[5,5]
#> [1] 0.2505698
attributes(diffAggreg.mtx)$matrices$agg[5,5] + correctedFact.num
#> [1] 0.2505698
mean(unlist(quantif.num)) + correctedFact.num
#> [1] 0.2505698
```

9. Differential at Aggregated level between Aggregation Classic Scaled and Control  


```r
attributes(diffAggreg.mtx)$matrices$agg[5,5]-attributes(diffAggreg.mtx)$matrices$aggCtrl[5,5] + correctedFact.num
#> [1] 0.0342001
attributes(diffAggreg.mtx)$matrices$aggCorrectedDelta[5,5]
#> [1] 0.0342001
mean(unlist(quantif.num)) - mean(unlist(quantif_ctrl.num)) + correctedFact.num
#> [1] 0.0342001
```
-->
</div>
</div>
<div class="section level3">
<h3 id="tips-5">Tips<a class="anchor" aria-label="anchor" href="#tips-5"></a>
</h3>
<ol style="list-style-type: decimal">
<li>When you perform a one sample aggregation, you could use any of
<code>matrices.lst</code> or <code>ctrlMatrices.lst</code>
parameters</li>
<li>If <code>rm0.bln</code> is <code>TRUE</code> all zeros in matrices
list are replace by NA.</li>
<li>It is possible to filter submatrices list by minimal or maximal
distance during the aggregation function.</li>
<li>Since statistical test is highly memory consumming, it is
recommended not to apply it each time.</li>
<li>Aggregation() on one sample keeps attributes of the matrices list
and add some news:</li>
</ol>
<ul>
<li>totalMatrixNumber: total number of matrices.<br>
</li>
<li>filteredMatrixNumber: number of matrices after distance
filtering.<br>
</li>
<li>minimalDistance: minimal distance between anchor and bait.<br>
</li>
<li>maximalDistance: maximal distance between anchor and bait.<br>
</li>
<li>aggregationMethod: The function apply to obtain the
aggregation.<br>
</li>
<li>zeroRemoved: A Boolean that indicate if zeros have been replaced by
NA.<br>
</li>
</ul>
<ol start="6" style="list-style-type: decimal">
<li>Aggregation() on two samples adds additional attributes:</li>
</ol>
<ul>
<li>correctedFact: The value that is add to the condition to delete some
noise. It’s compute by the median difference between condition and
control in an background area (e.g upper right corner in matrices).</li>
<li>matrices: The list of matrices.
<ul>
<li>agg: Aggregation of the condition.<br>
</li>
<li>aggCtrl: Aggregation of the control.<br>
</li>
<li>aggCorrected: Aggregation of the condition corrected with
correctedFact.<br>
</li>
<li>aggDelta: the difference between the aggregated matrix of the
condition and the aggregated matrix of the control.<br>
</li>
<li>aggCorrectedDelta: the difference between the aggregated matrix of
the condition corrected with correctedFact and the aggregated matrix of
the control.</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="aggregations-plots">Aggregations plots<a class="anchor" aria-label="anchor" href="#aggregations-plots"></a>
</h2>
<div class="section level3">
<h3 id="preparation-of-aggregated-matrices">Preparation of aggregated matrices<a class="anchor" aria-label="anchor" href="#preparation-of-aggregated-matrices"></a>
</h3>
<ol style="list-style-type: decimal">
<li>Control aggregation</li>
</ol>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">oriented_interactions_RFmatrix_ctrl.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OrienteMatrix.html">OrienteMatrix</a></span><span class="op">(</span><span class="va">interactions_RFmatrix_ctrl.lst</span><span class="op">)</span></span>
<span><span class="va">aggreg.mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Aggregation.html">Aggregation</a></span><span class="op">(</span></span>
<span>    ctrlMatrices.lst<span class="op">=</span><span class="va">oriented_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>    agg.fun<span class="op">=</span><span class="st">"mean"</span>,</span>
<span>    rm0.bln<span class="op">=</span><span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">aggreg.mtx</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">31</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">aggreg.mtx</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">31</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Differential aggregation</li>
</ol>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">oriented_interactions_RFmatrix.lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OrienteMatrix.html">OrienteMatrix</a></span><span class="op">(</span><span class="va">interactions_RFmatrix.lst</span><span class="op">)</span></span>
<span><span class="va">diffAggreg.mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Aggregation.html">Aggregation</a></span><span class="op">(</span></span>
<span>    ctrlMatrices.lst    <span class="op">=</span> <span class="va">oriented_interactions_RFmatrix_ctrl.lst</span>,</span>
<span>    matrices.lst        <span class="op">=</span> <span class="va">oriented_interactions_RFmatrix.lst</span>,</span>
<span>    minDist             <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    maxDist             <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    agg.fun             <span class="op">=</span> <span class="st">"mean"</span>,</span>
<span>    rm0.bln             <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    diff.fun            <span class="op">=</span> <span class="st">"log2+1"</span>,</span>
<span>    scaleCorrection.bln <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    correctionArea.lst  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span> i<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span> , j<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">72</span><span class="op">:</span><span class="fl">101</span><span class="op">)</span> <span class="op">)</span>,</span>
<span>    statCompare.bln     <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">diffAggreg.mtx</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">31</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">diffAggreg.mtx</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">31</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"*"</span>,<span class="fl">2</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">""</span>,<span class="fl">33</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plots">Plots<a class="anchor" aria-label="anchor" href="#plots"></a>
</h3>
<div class="section level4">
<h4 id="simple-aggregation-plot">Simple aggregation plot:<a class="anchor" aria-label="anchor" href="#simple-aggregation-plot"></a>
</h4>
<p>ggAPA() function creates a ggplot object (ggplot2::geom_raster)</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx   <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr <span class="op">=</span> <span class="st">"APA"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-136-1.png" width="672"></p>
</div>
<div class="section level4">
<h4 id="further-visualisation-parameters">Further visualisation parameters:<a class="anchor" aria-label="anchor" href="#further-visualisation-parameters"></a>
</h4>
<div class="section level5">
<h5 id="trimming-aggregated-values-for-visualisation">Trimming aggregated values for visualisation:<a class="anchor" aria-label="anchor" href="#trimming-aggregated-values-for-visualisation"></a>
</h5>
<p>You could remove a percentage of value choose in upper tail, lower
tail or both tails of the distribution.</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx      <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr    <span class="op">=</span> <span class="st">"APA 30% trimmed on upper side"</span>,</span>
<span>    trimPrct.num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>    bounds.chr   <span class="op">=</span> <span class="st">"upper"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in max(unlist(bounds.num_lst[1]), na.rm = TRUE): no non-missing</span></span>
<span><span class="co">#&gt; arguments to max; returning -Inf</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-137-1.png" width="672"></p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx      <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr    <span class="op">=</span> <span class="st">"APA 30% trimmed on upper side"</span>,</span>
<span>    trimPrct.num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>    bounds.chr   <span class="op">=</span> <span class="st">"lower"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in min(unlist(bounds.num_lst[2]), na.rm = TRUE): no non-missing</span></span>
<span><span class="co">#&gt; arguments to min; returning Inf</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-137-2.png" width="672"></p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx      <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr    <span class="op">=</span> <span class="st">"APA 30% trimmed"</span>,</span>
<span>    trimPrct.num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>    bounds.chr   <span class="op">=</span> <span class="st">"both"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-137-3.png" width="672"></p>
</div>
<div class="section level5">
<h5 id="modifying-color-scale">Modifying color scale:<a class="anchor" aria-label="anchor" href="#modifying-color-scale"></a>
</h5>
<div class="section level6">
<h6 id="min-and-max-color-scale">Min and max color scale<a class="anchor" aria-label="anchor" href="#min-and-max-color-scale"></a>
</h6>
<p>Example of user-defined min and max color scale</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx         <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr       <span class="op">=</span> <span class="st">"APA [.15-.35]"</span>,</span>
<span>    minBoundary.num <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>    maxBoundary.num <span class="op">=</span> <span class="fl">0.35</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-138-1.png" width="672"></p>
</div>
<div class="section level6">
<h6 id="center-color-scale">Center color scale<a class="anchor" aria-label="anchor" href="#center-color-scale"></a>
</h6>
<p>Example of user-define color scale center</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx    <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr  <span class="op">=</span> <span class="st">"APA center on 0.2"</span>,</span>
<span>    center.num <span class="op">=</span> <span class="fl">0.2</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-139-1.png" width="672"></p>
</div>
<div class="section level6">
<h6 id="change-color-breaks">Change color breaks<a class="anchor" aria-label="anchor" href="#change-color-breaks"></a>
</h6>
<p>Exmaples of user-defined color breaks</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx       <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr     <span class="op">=</span> <span class="st">"APA [.1, .2, .3, .4]"</span>,</span>
<span>    colBreaks.num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>,<span class="fl">0.2</span>,<span class="fl">0.3</span>,<span class="fl">0.4</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-140-1.png" width="672"></p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx       <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr     <span class="op">=</span> <span class="st">"APA [.10, .13, .16, .20, .30, .40]"</span>,</span>
<span>    colBreaks.num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>,<span class="fl">0.13</span>,<span class="fl">0.16</span>,<span class="fl">0.2</span>,<span class="fl">0.3</span>,<span class="fl">0.4</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-140-2.png" width="672"></p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx       <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr     <span class="op">=</span> <span class="st">"APA [.10, .20, .30, .33, .36, .40]"</span>,</span>
<span>    colBreaks.num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>,<span class="fl">0.2</span>,<span class="fl">0.3</span>,<span class="fl">0.33</span>,<span class="fl">0.36</span>,<span class="fl">0.40</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-140-3.png" width="672"></p>
</div>
<div class="section level6">
<h6 id="change-color-scale-bias">Change color scale bias<a class="anchor" aria-label="anchor" href="#change-color-scale-bias"></a>
</h6>
<p>Examples of different color scale biais.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx    <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr  <span class="op">=</span> <span class="st">"APA"</span>,</span>
<span>    colorScale <span class="op">=</span> <span class="st">"density"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-141-1.png" width="672"></p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx     <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr   <span class="op">=</span> <span class="st">"APA"</span>,</span>
<span>    bias.num    <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-141-2.png" width="672"></p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx     <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr   <span class="op">=</span> <span class="st">"APA"</span>,</span>
<span>    bias.num    <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-141-3.png" width="672"></p>
</div>
<div class="section level6">
<h6 id="changed-color">Changed color<a class="anchor" aria-label="anchor" href="#changed-color"></a>
</h6>
<p>here is an option to change the color of heatmap and color of NA
values.</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx     <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr   <span class="op">=</span> <span class="st">"APA"</span>,</span>
<span>    heatmap.col <span class="op">=</span> <span class="fu">SuperTK</span><span class="fu">::</span><span class="fu"><a href="https://nchanard.github.io/ColorTK/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span>,</span>
<span>    na.col      <span class="op">=</span> <span class="st">"black"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-142-1.png" width="672"></p>
</div>
</div>
<div class="section level5">
<h5 id="blurred-visualisation">Blurred visualisation<a class="anchor" aria-label="anchor" href="#blurred-visualisation"></a>
</h5>
<p>There is an option to apply a blurr on the heatmap that could remove
some noise.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx           <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr         <span class="op">=</span> <span class="st">"APA"</span>,</span>
<span>    blurPass.num      <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    blurSd.num        <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>    lowerTri.num <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-143-1.png" width="672"></p>
</div>
<div class="section level5">
<h5 id="ggplot-object-modifications">ggplot object modifications<a class="anchor" aria-label="anchor" href="#ggplot-object-modifications"></a>
</h5>
<p>Since ggAPA() returns a ggplot object, it is possible to modify it
following the ggplot2 grammar</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ggAPA.html">ggAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx     <span class="op">=</span> <span class="va">aggreg.mtx</span>,</span>
<span>    title.chr   <span class="op">=</span> <span class="st">"APA"</span>,</span>
<span><span class="op">)</span> <span class="op">+</span> </span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    title    <span class="op">=</span> <span class="st">"New title"</span>,</span>
<span>    subtitle <span class="op">=</span> <span class="st">"and subtitle"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-144-1.png" width="672"></p>
</div>
</div>
<div class="section level4">
<h4 id="multiplot-aggregation-function">Multiplot aggregation function:<a class="anchor" aria-label="anchor" href="#multiplot-aggregation-function"></a>
</h4>
<p>PlotAPA() function plots multiple pages of aggregated matrices with
different visualisation parameters.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/PlotAPA.html">PlotAPA</a></span><span class="op">(</span></span>
<span>    apa.mtx                  <span class="op">=</span> <span class="va">diffAggreg.mtx</span>,</span>
<span>    trimPrct.num             <span class="op">=</span> <span class="fl">20</span>,</span>
<span>    minBoundary.num          <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>,</span>
<span>    center.num               <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    maxBoundary.num          <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    minConditionBoundary.num <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    maxConditionBoundary.num <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre></div>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-4.png" width="700"><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-5.png" width="700"><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-6.png" width="700"><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-7.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-8.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-9.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-10.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-11.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-12.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-13.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-14.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-15.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-16.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-17.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-18.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-19.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-20.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-21.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-22.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-23.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `data.dtf$x` is discouraged. Use `x` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$j` is discouraged. Use `j` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `data.dtf$i` is discouraged. Use `i` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-24.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `mu.tbl$grp.mean` is discouraged. Use `grp.mean` instead.</span></span>
<span><span class="co">#&gt; Warning: Use of `mu.tbl$class` is discouraged. Use `class` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-25.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `mu.tbl$grp.mean` is discouraged. Use `grp.mean` instead.</span></span>
<span><span class="co">#&gt; Use of `mu.tbl$class` is discouraged. Use `class` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-26.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `mu.tbl$grp.mean` is discouraged. Use `grp.mean` instead.</span></span>
<span><span class="co">#&gt; Use of `mu.tbl$class` is discouraged. Use `class` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-27.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Warning: Use of `mu.tbl$grp.mean` is discouraged. Use `grp.mean` instead.</span></span>
<span><span class="co">#&gt; Use of `mu.tbl$class` is discouraged. Use `class` instead.</span></span></code></pre>
<p><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-28.png" width="700"><img src="Genomic2DTK_files/figure-html/unnamed-chunk-145-29.png" width="700"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicolas Chanard .</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
